diff -urN scip-9.2.1-orig/src/cppad/core/check_for_nan.hpp scip-9.2.1/src/cppad/core/check_for_nan.hpp
--- scip-9.2.1-orig/src/cppad/core/check_for_nan.hpp	2025-01-29 19:38:36.000000000 +0800
+++ scip-9.2.1/src/cppad/core/check_for_nan.hpp	2025-03-29 20:08:43.052326300 +0800
@@ -137,7 +137,9 @@
 
 # if CPPAD_HAS_MKSTEMP
 # include <stdlib.h>
+# ifndef _WIN32
 # include <unistd.h>
+# endif
 # else
 # if CPPAD_HAS_TMPNAM_S
 # include <stdio.h>
diff -urN scip-9.2.1-orig/src/scip/githash.c scip-9.2.1/src/scip/githash.c
--- scip-9.2.1-orig/src/scip/githash.c	2025-01-29 19:39:03.000000000 +0800
+++ scip-9.2.1/src/scip/githash.c	2025-03-29 20:55:41.999091900 +0800
@@ -1 +1 @@
-#define SCIP_GITHASH "0d2d3c7c2d"
+#define SCIP_GITHASH "0d2d3c7c2d"
diff -urN scip-9.2.1-orig/src/scip/reader_zpl.c scip-9.2.1/src/scip/reader_zpl.c
--- scip-9.2.1-orig/src/scip/reader_zpl.c	2025-01-29 19:38:39.000000000 +0800
+++ scip-9.2.1/src/scip/reader_zpl.c	2025-03-29 20:09:11.899700700 +0800
@@ -35,7 +35,9 @@
 
 #ifdef SCIP_WITH_ZIMPL
 
+#ifndef _WIN32
 #include <unistd.h>
+#endif
 #include <stdbool.h>
 #include <string.h>
 
