diff -urN pixman-0.44.2-orig/demos/meson.build pixman-0.44.2/demos/meson.build
--- pixman-0.44.2-orig/demos/meson.build	2024-12-03 11:51:09.000000000 +0800
+++ pixman-0.44.2/demos/meson.build	2025-04-12 00:20:22.541886200 +0800
@@ -48,6 +48,7 @@
     'demo',
     ['gtk-utils.c', config_h, version_h],
     dependencies : [libtestutils_dep, dep_gtk, dep_glib, dep_png, dep_m, dep_openmp],
+    name_suffix : 'lib',
     include_directories : inc_pixman,
   )
 
diff -urN pixman-0.44.2-orig/pixman/meson.build pixman-0.44.2/pixman/meson.build
--- pixman-0.44.2-orig/pixman/meson.build	2024-12-03 11:51:09.000000000 +0800
+++ pixman-0.44.2/pixman/meson.build	2025-04-12 07:15:34.605556600 +0800
@@ -66,7 +66,8 @@
     pixman_simd_libs += static_library(
       name,
       [name + '.c', config_h, version_h, simd[3]],
-      c_args : simd[2]
+      c_args : simd[2],
+      name_suffix : 'lib'
     )
   endif
 endforeach
diff -urN pixman-0.44.2-orig/test/utils/meson.build pixman-0.44.2/test/utils/meson.build
--- pixman-0.44.2-orig/test/utils/meson.build	2024-12-03 11:51:09.000000000 +0800
+++ pixman-0.44.2/test/utils/meson.build	2025-04-12 00:20:22.541886200 +0800
@@ -22,6 +22,7 @@
   'testutils',
   ['utils.c', 'utils-prng.c', config_h],
   dependencies : [idep_pixman, dep_openmp, dep_m, dep_png],
+  name_suffix : 'lib',
 )
 
 libtestutils_dep = declare_dependency(
