diff -urN pkg-config-0.29.2-orig/glib/glib/glib-init.c pkg-config-0.29.2/glib/glib/glib-init.c
--- pkg-config-0.29.2-orig/glib/glib/glib-init.c	2016-04-12 05:39:26.000000000 +0800
+++ pkg-config-0.29.2/glib/glib/glib-init.c	2025-02-10 11:26:32.709548700 +0800
@@ -106,7 +106,11 @@
    * inside GLib.
    */
 
+#ifdef _MSC_VER
+  if (!_stricmp (string, "help"))
+#else
   if (!strcasecmp (string, "help"))
+#endif
     {
       /* using stdio directly for the reason stated above */
       fprintf (stderr, "Supported debug values:");
