diff -urN gobject-introspection-1.82.0-orig/girepository/cmph/meson.build gobject-introspection-1.82.0/girepository/cmph/meson.build
--- gobject-introspection-1.82.0-orig/girepository/cmph/meson.build	2024-09-13 21:08:54.000000000 +0800
+++ gobject-introspection-1.82.0/girepository/cmph/meson.build	2025-04-12 07:43:56.875265000 +0800
@@ -50,6 +50,7 @@
 cmph = static_library('cmph',
   sources: cmph_sources,
   c_args: gi_hidden_visibility_cflags + custom_c_args,
+  name_suffix: 'lib',
   dependencies: cmph_deps,
 )
 
diff -urN gobject-introspection-1.82.0-orig/girepository/meson.build gobject-introspection-1.82.0/girepository/meson.build
--- gobject-introspection-1.82.0-orig/girepository/meson.build	2024-09-13 21:08:54.000000000 +0800
+++ gobject-introspection-1.82.0/girepository/meson.build	2025-04-12 07:44:10.513152600 +0800
@@ -15,6 +15,7 @@
   sources: 'gthash.c',
   include_directories : configinc,
   c_args: gi_hidden_visibility_cflags + custom_c_args,
+  name_suffix: 'lib',
   dependencies: [
     cmph_dep,
     glib_dep,
@@ -47,6 +48,7 @@
   ],
   c_args: gi_hidden_visibility_cflags + custom_c_args,
   include_directories : configinc,
+  name_suffix: 'lib',
   dependencies: [girepo_gthash_dep, libffi_dep],
 )
 
diff -urN gobject-introspection-1.82.0-orig/giscanner/meson.build gobject-introspection-1.82.0/giscanner/meson.build
--- gobject-introspection-1.82.0-orig/giscanner/meson.build	2024-09-13 21:08:54.000000000 +0800
+++ gobject-introspection-1.82.0/giscanner/meson.build	2025-04-12 07:44:17.967061500 +0800
@@ -93,6 +93,7 @@
   c_args: giscanner_args + custom_c_args,
   include_directories : configinc,
   dependencies: [glib_dep, gobject_dep, gio_dep, gmodule_dep],
+  name_suffix: 'lib',
 )
 
 # https://github.com/mesonbuild/meson/issues/4117
