diff -urN lapackpp-2024.10.26-orig/src/lartg.cc lapackpp-2024.10.26/src/lartg.cc
--- lapackpp-2024.10.26-orig/src/lartg.cc	2024-10-26 23:17:25.000000000 +0800
+++ lapackpp-2024.10.26/src/lartg.cc	2024-12-02 11:12:53.770364500 +0800
@@ -3,6 +3,7 @@
 // This program is free software: you can redistribute it and/or modify it under
 // the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
 
+#include "lapack_internal.hh"
 #include "lapack/fortran.h"
 
 namespace lapack {
