diff -urN mesa-25.0.2-orig/src/amd/addrlib/meson.build mesa-25.0.2/src/amd/addrlib/meson.build
--- mesa-25.0.2-orig/src/amd/addrlib/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/amd/addrlib/meson.build	2025-03-31 13:58:23.555617100 +0800
@@ -80,4 +80,5 @@
   ],
   cpp_args : cpp_args_addrlib,
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/amd/common/meson.build mesa-25.0.2/src/amd/common/meson.build
--- mesa-25.0.2-orig/src/amd/common/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/amd/common/meson.build	2025-03-31 13:58:23.558611700 +0800
@@ -162,7 +162,8 @@
   ],
   link_with: [ link_with ],
   gnu_symbol_visibility : 'hidden',
-  c_args : c_args
+  c_args : c_args,
+  name_suffix : 'lib',
 )
 
 idep_amdgfxregs_h = declare_dependency(sources : [amdgfxregs_h])
diff -urN mesa-25.0.2-orig/src/amd/compiler/meson.build mesa-25.0.2/src/amd/compiler/meson.build
--- mesa-25.0.2-orig/src/amd/compiler/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/amd/compiler/meson.build	2025-03-31 13:58:23.562598800 +0800
@@ -84,6 +84,7 @@
   gnu_symbol_visibility : 'hidden',
   build_by_default : true,
   cpp_args : [cpp_args_aco, cpp_msvc_compat_args],
+  name_suffix : 'lib', 
 )
 
 # Also link with aco
diff -urN mesa-25.0.2-orig/src/amd/llvm/meson.build mesa-25.0.2/src/amd/llvm/meson.build
--- mesa-25.0.2-orig/src/amd/llvm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/amd/llvm/meson.build	2025-03-31 13:58:23.566269100 +0800
@@ -26,5 +26,6 @@
     idep_nir_headers, idep_amdgfxregs_h,
   ],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',
 )
 
diff -urN mesa-25.0.2-orig/src/amd/vpelib/meson.build mesa-25.0.2/src/amd/vpelib/meson.build
--- mesa-25.0.2-orig/src/amd/vpelib/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/amd/vpelib/meson.build	2025-03-31 13:58:23.571255500 +0800
@@ -143,5 +143,6 @@
   vpe_files,
   install : false,
   c_args : c_args_vpe,
-  include_directories : inc_amd_vpe
+  include_directories : inc_amd_vpe,
+  name_suffix : 'lib',   
 )
diff -urN mesa-25.0.2-orig/src/asahi/compiler/meson.build mesa-25.0.2/src/asahi/compiler/meson.build
--- mesa-25.0.2-orig/src/asahi/compiler/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/asahi/compiler/meson.build	2025-03-31 13:58:23.577977000 +0800
@@ -99,6 +99,7 @@
   c_args : [no_override_init_args, '-Wno-c2x-extensions'],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib', 
 )
 
 if with_tests
diff -urN mesa-25.0.2-orig/src/asahi/layout/meson.build mesa-25.0.2/src/asahi/layout/meson.build
--- mesa-25.0.2-orig/src/asahi/layout/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/asahi/layout/meson.build	2025-03-31 13:58:23.583283200 +0800
@@ -15,6 +15,7 @@
   c_args : [no_override_init_args],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib', 
 )
 
 if with_tests
diff -urN mesa-25.0.2-orig/src/asahi/lib/meson.build mesa-25.0.2/src/asahi/lib/meson.build
--- mesa-25.0.2-orig/src/asahi/lib/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/asahi/lib/meson.build	2025-03-31 13:58:23.589266400 +0800
@@ -40,6 +40,7 @@
   c_args : [no_override_init_args, '-Wno-c2x-extensions'],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 libasahi_decode_shared = shared_library(
@@ -73,6 +74,7 @@
   link_with: [libasahi_decode, libvdrm],
   dependencies: [dep_libdrm, dep_valgrind, idep_nir, idep_mesautil, idep_libagx],
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 if with_tests
diff -urN mesa-25.0.2-orig/src/asahi/vulkan/meson.build mesa-25.0.2/src/asahi/vulkan/meson.build
--- mesa-25.0.2-orig/src/asahi/vulkan/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/asahi/vulkan/meson.build	2025-03-31 13:58:23.593771400 +0800
@@ -93,6 +93,7 @@
   c_args : ['-Wno-c2x-extensions'],
   dependencies : [hk_deps],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',  
 )
 
 libvulkan_asahi = shared_library(
diff -urN mesa-25.0.2-orig/src/broadcom/cle/meson.build mesa-25.0.2/src/broadcom/cle/meson.build
--- mesa-25.0.2-orig/src/broadcom/cle/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/broadcom/cle/meson.build	2025-03-31 13:58:23.596762400 +0800
@@ -47,4 +47,5 @@
   gnu_symbol_visibility : 'hidden',
   dependencies : [dep_libdrm, dep_valgrind, dep_expat, dep_zlib],
   build_by_default : false,
+  name_suffix : 'lib',   
 )
diff -urN mesa-25.0.2-orig/src/broadcom/compiler/meson.build mesa-25.0.2/src/broadcom/compiler/meson.build
--- mesa-25.0.2-orig/src/broadcom/compiler/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/broadcom/compiler/meson.build	2025-03-31 13:58:23.599754100 +0800
@@ -46,6 +46,7 @@
   gnu_symbol_visibility : 'hidden',
   dependencies : [dep_libdrm, dep_valgrind, idep_nir_headers, idep_mesautil],
   build_by_default : false,
+  name_suffix : 'lib', 
 )
 
 v3d_libs += libbroadcom_compiler
diff -urN mesa-25.0.2-orig/src/broadcom/ds/meson.build mesa-25.0.2/src/broadcom/ds/meson.build
--- mesa-25.0.2-orig/src/broadcom/ds/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/broadcom/ds/meson.build	2025-03-31 13:58:23.602912700 +0800
@@ -20,6 +20,7 @@
     idep_broadcom_perfcntrs,
   ],
   c_args: [no_override_init_args],
+  name_suffix : 'lib', 
 )
 
 pps_broadcom_dep = declare_dependency(
diff -urN mesa-25.0.2-orig/src/broadcom/meson.build mesa-25.0.2/src/broadcom/meson.build
--- mesa-25.0.2-orig/src/broadcom/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/broadcom/meson.build	2025-03-31 13:58:23.605718100 +0800
@@ -37,6 +37,7 @@
     c_args : [no_override_init_args, '-DV3D_VERSION=' + ver],
     gnu_symbol_visibility : 'hidden',
     dependencies: [dep_valgrind, dep_thread],
+    name_suffix : 'lib',   
   )
 endforeach
 
@@ -56,6 +57,7 @@
   c_args : [v3d_args, v3d_neon_c_args],
   gnu_symbol_visibility : 'hidden',
   dependencies : [dep_libdrm, dep_valgrind, idep_nir_headers, idep_mesautil],
+  name_suffix : 'lib',
 )
 
 libbroadcom_v3d = static_library(
@@ -71,6 +73,7 @@
   link_with: [libv3d_neon],
   build_by_default : false,
   dependencies: [dep_valgrind, dep_thread, idep_mesautil, dep_libdrm],
+  name_suffix : 'lib', 
 )
 
 subdir('perfcntrs')
diff -urN mesa-25.0.2-orig/src/broadcom/perfcntrs/meson.build mesa-25.0.2/src/broadcom/perfcntrs/meson.build
--- mesa-25.0.2-orig/src/broadcom/perfcntrs/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/broadcom/perfcntrs/meson.build	2025-03-31 13:58:23.608779900 +0800
@@ -20,6 +20,7 @@
     c_args : [v3d_simulator_arg, '-DV3D_VERSION=' + ver],
     gnu_symbol_visibility: 'hidden',
     dependencies : [dep_v3d_hw, dep_libdrm, dep_valgrind, idep_mesautil],
+    name_suffix : 'lib',   
 )
 endforeach
 
@@ -32,6 +33,7 @@
   dependencies : [dep_v3d_hw, dep_libdrm, dep_valgrind, idep_mesautil],
   link_with : [per_version_libs],
   build_by_default : false,
+  name_suffix : 'lib', 
 )
 
 idep_broadcom_perfcntrs = declare_dependency(
diff -urN mesa-25.0.2-orig/src/broadcom/qpu/meson.build mesa-25.0.2/src/broadcom/qpu/meson.build
--- mesa-25.0.2-orig/src/broadcom/qpu/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/broadcom/qpu/meson.build	2025-03-31 13:58:23.611769900 +0800
@@ -15,6 +15,7 @@
   gnu_symbol_visibility : 'hidden',
   dependencies : [dep_libdrm, dep_valgrind],
   build_by_default : false,
+  name_suffix : 'lib', 
 )
 
 v3d_libs += libbroadcom_qpu
diff -urN mesa-25.0.2-orig/src/broadcom/simulator/meson.build mesa-25.0.2/src/broadcom/simulator/meson.build
--- mesa-25.0.2-orig/src/broadcom/simulator/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/broadcom/simulator/meson.build	2025-03-31 13:58:23.615760000 +0800
@@ -20,6 +20,7 @@
     c_args : [v3d_simulator_arg, '-DV3D_VERSION=' + ver],
     gnu_symbol_visibility: 'hidden',
     dependencies : [dep_v3d_hw, dep_libdrm, dep_valgrind],
+    name_suffix : 'lib', 
 )
 endforeach
 
@@ -33,6 +34,7 @@
   dependencies : [dep_v3d_hw, dep_libdrm, dep_valgrind],
   link_with : [per_version_libs],
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 v3d_libs += libbroadcom_simulator
diff -urN mesa-25.0.2-orig/src/broadcom/vulkan/meson.build mesa-25.0.2/src/broadcom/vulkan/meson.build
--- mesa-25.0.2-orig/src/broadcom/vulkan/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/broadcom/vulkan/meson.build	2025-03-31 13:58:23.619461000 +0800
@@ -94,6 +94,7 @@
     c_args : [v3dv_flags, '-DV3D_VERSION=' + ver],
     gnu_symbol_visibility : 'hidden',
     dependencies : [v3dv_deps],
+    name_suffix : 'lib',
 )
 endforeach
 
diff -urN mesa-25.0.2-orig/src/c11/impl/meson.build mesa-25.0.2/src/c11/impl/meson.build
--- mesa-25.0.2-orig/src/c11/impl/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/c11/impl/meson.build	2025-03-31 13:58:23.622453400 +0800
@@ -22,7 +22,8 @@
   include_directories : inc_util_c11,
   c_args : [c_msvc_compat_args],
   gnu_symbol_visibility : 'hidden',
-  build_by_default : false
+  build_by_default : false,
+  name_suffix : 'lib',
 )
 
 idep_mesautilc11 = declare_dependency(
diff -urN mesa-25.0.2-orig/src/compiler/clc/meson.build mesa-25.0.2/src/compiler/clc/meson.build
--- mesa-25.0.2-orig/src/compiler/clc/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/compiler/clc/meson.build	2025-03-31 13:58:23.625792400 +0800
@@ -106,7 +106,8 @@
   c_args : _libmesaclc_c_args,
   cpp_args : [_libmesaclc_cpp_args, _libmesaclc_c_args],
   dependencies: [idep_nir, dep_clang, dep_llvm, dep_llvmspirvlib,
-                 idep_mesautil, dep_spirv_tools, idep_vtn, dep_version]
+                 idep_mesautil, dep_spirv_tools, idep_vtn, dep_version],
+  name_suffix : 'lib',
 )
 
 _idep_mesaclc_link_args = []
diff -urN mesa-25.0.2-orig/src/compiler/glsl/glcpp/meson.build mesa-25.0.2/src/compiler/glsl/glcpp/meson.build
--- mesa-25.0.2-orig/src/compiler/glsl/glcpp/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/compiler/glsl/glcpp/meson.build	2025-03-31 13:58:23.628130800 +0800
@@ -37,6 +37,7 @@
   cpp_args : [cpp_msvc_compat_args],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 libglcpp_standalone = static_library(
@@ -49,6 +50,7 @@
   cpp_args : [cpp_msvc_compat_args],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 glcpp = executable(
diff -urN mesa-25.0.2-orig/src/compiler/glsl/meson.build mesa-25.0.2/src/compiler/glsl/meson.build
--- mesa-25.0.2-orig/src/compiler/glsl/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/compiler/glsl/meson.build	2025-03-31 13:58:23.632119900 +0800
@@ -8,6 +8,7 @@
   dependencies : [dep_valgrind, idep_mesautil],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 subdir('glcpp')
@@ -218,6 +219,7 @@
   include_directories : [inc_include, inc_src, inc_mapi, inc_mesa, inc_gallium, inc_gallium_aux],
   dependencies : [idep_nir, idep_mesautil],
   build_by_default : false,
+  name_suffix : 'lib',  
 )
 
 idep_libglsl = declare_dependency(
@@ -235,6 +237,7 @@
   link_with : [libglsl, libglsl_util, libglcpp_standalone],
   dependencies : [idep_mesautil, idep_getopt, idep_compiler, idep_nir],
   build_by_default : false,
+  name_suffix : 'lib', 
 )
 
 glsl_compiler = executable(
diff -urN mesa-25.0.2-orig/src/compiler/isaspec/meson.build mesa-25.0.2/src/compiler/isaspec/meson.build
--- mesa-25.0.2-orig/src/compiler/isaspec/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/compiler/isaspec/meson.build	2025-03-31 13:58:23.636109000 +0800
@@ -13,4 +13,5 @@
   ['isaspec.c'],
   dependencies: [idep_mesautil],
   gnu_symbol_visibility: 'hidden',
+  name_suffix : 'lib', 
 )
diff -urN mesa-25.0.2-orig/src/compiler/meson.build mesa-25.0.2/src/compiler/meson.build
--- mesa-25.0.2-orig/src/compiler/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/compiler/meson.build	2025-03-31 13:58:23.638102900 +0800
@@ -51,6 +51,7 @@
   gnu_symbol_visibility : 'hidden',
   dependencies : [dep_valgrind],
   build_by_default : false,
+  name_suffix : 'lib', 
 )
 
 idep_compiler = declare_dependency(
diff -urN mesa-25.0.2-orig/src/compiler/nir/meson.build mesa-25.0.2/src/compiler/nir/meson.build
--- mesa-25.0.2-orig/src/compiler/nir/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/compiler/nir/meson.build	2025-03-31 13:58:23.641612100 +0800
@@ -325,6 +325,7 @@
   gnu_symbol_visibility : 'hidden',
   dependencies : [idep_compiler, dep_valgrind, idep_mesautil],
   build_by_default : false,
+  name_suffix : 'lib', 
 )
 
 inc_nir = include_directories('.')
diff -urN mesa-25.0.2-orig/src/compiler/rust/meson.build mesa-25.0.2/src/compiler/rust/meson.build
--- mesa-25.0.2-orig/src/compiler/rust/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/compiler/rust/meson.build	2025-03-31 13:58:23.645863300 +0800
@@ -61,6 +61,7 @@
   include_directories : [inc_include, inc_util],
   c_args : [no_override_init_args],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib', 
 )
 
 _idep_libcompiler_c = declare_dependency(
@@ -99,6 +100,7 @@
   gnu_symbol_visibility : 'hidden',
   rust_abi : 'rust',
   dependencies: [_idep_libcompiler_c],
+  name_suffix : 'lib',  
 )
 
 # TODO: Linking Rust executables (such as unit tests) doesn't play nicely
@@ -135,6 +137,7 @@
   dependencies : [dep_syn],
   rust_abi : 'rust',
   native : true,
+  name_suffix : 'lib', 
 )
 
 idep_compiler_proc_rs = declare_dependency(
diff -urN mesa-25.0.2-orig/src/compiler/spirv/meson.build mesa-25.0.2/src/compiler/spirv/meson.build
--- mesa-25.0.2-orig/src/compiler/spirv/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/compiler/spirv/meson.build	2025-03-31 13:58:23.649329000 +0800
@@ -57,6 +57,7 @@
   gnu_symbol_visibility : 'hidden',
   dependencies : [dep_spirv_tools, idep_nir, dep_valgrind],
   build_by_default : false,
+  name_suffix : 'lib',  
 )
 
 idep_vtn = declare_dependency(
diff -urN mesa-25.0.2-orig/src/drm-shim/meson.build mesa-25.0.2/src/drm-shim/meson.build
--- mesa-25.0.2-orig/src/drm-shim/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/drm-shim/meson.build	2025-03-31 13:58:23.653342500 +0800
@@ -11,6 +11,7 @@
   include_directories: [inc_include, inc_src],
   dependencies: [dep_libdrm, idep_mesautil, dep_dl],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',
 )
 dep_drm_shim = declare_dependency(
   link_with: drm_shim,
diff -urN mesa-25.0.2-orig/src/egl/wayland/wayland-drm/meson.build mesa-25.0.2/src/egl/wayland/wayland-drm/meson.build
--- mesa-25.0.2-orig/src/egl/wayland/wayland-drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/egl/wayland/wayland-drm/meson.build	2025-03-31 13:58:23.656334200 +0800
@@ -30,6 +30,7 @@
   gnu_symbol_visibility : 'hidden',
   dependencies : [dep_wayland_server],
   build_by_default : false,
+  name_suffix : 'lib', 
 )
 
 # linux-dmabuf isn't part of wayland-drm, but this happens to be the only
diff -urN mesa-25.0.2-orig/src/etnaviv/drm/meson.build mesa-25.0.2/src/etnaviv/drm/meson.build
--- mesa-25.0.2-orig/src/etnaviv/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/etnaviv/drm/meson.build	2025-03-31 13:58:23.659450500 +0800
@@ -33,6 +33,7 @@
     libetnaviv_hwdb
   ],
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 if with_tools.contains('etnaviv')
diff -urN mesa-25.0.2-orig/src/etnaviv/hwdb/meson.build mesa-25.0.2/src/etnaviv/hwdb/meson.build
--- mesa-25.0.2-orig/src/etnaviv/hwdb/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/etnaviv/hwdb/meson.build	2025-03-31 13:58:23.662418800 +0800
@@ -39,4 +39,5 @@
   c_args : [no_override_init_args],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib', 
 )
diff -urN mesa-25.0.2-orig/src/etnaviv/isa/meson.build mesa-25.0.2/src/etnaviv/isa/meson.build
--- mesa-25.0.2-orig/src/etnaviv/isa/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/etnaviv/isa/meson.build	2025-03-31 13:58:23.666432400 +0800
@@ -32,6 +32,7 @@
     inc_src,
   ],
   gnu_symbol_visibility: 'hidden',
+  name_suffix : 'lib', 
 )
 
 idep_libetnaviv_decode = declare_dependency(
@@ -82,6 +83,7 @@
   include_directories: [
     inc_src,
   ],
+  name_suffix : 'lib',
 )
 
 disasm = executable(
@@ -119,6 +121,7 @@
     rust_args: [
       '-Anon_camel_case_types',
     ],
+    name_suffix : 'lib',
   )
 
   dep_indexmap = dependency('indexmap',
@@ -169,6 +172,7 @@
     rust_abi : 'c',
     dependencies: [dep_indexmap, dep_roxmltree, dep_pest],
     link_with: [_libetnaviv_isa_bindings_gen, _libetnaviv_isa_proc_rs],
+    name_suffix : 'lib',
   )
 
   etnaviv_assembler = executable(
@@ -185,6 +189,7 @@
   libetnaviv_isa_rs = static_library(
     'etnaviv_isa_rs',
     sources: [],
+    name_suffix : 'lib',
   )
 endif
 
diff -urN mesa-25.0.2-orig/src/freedreno/common/meson.build mesa-25.0.2/src/freedreno/common/meson.build
--- mesa-25.0.2-orig/src/freedreno/common/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/freedreno/common/meson.build	2025-03-31 13:58:23.669945600 +0800
@@ -35,6 +35,7 @@
   build_by_default : true,
   dependencies: [idep_mesautil],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib', 
 )
 
 idep_libfreedreno_common = declare_dependency(
diff -urN mesa-25.0.2-orig/src/freedreno/decode/meson.build mesa-25.0.2/src/freedreno/decode/meson.build
--- mesa-25.0.2-orig/src/freedreno/decode/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/freedreno/decode/meson.build	2025-03-31 13:58:23.673112200 +0800
@@ -35,6 +35,7 @@
     _libmesa_util,
   ],
   build_by_default: false,
+  name_suffix : 'lib',
 )
 
 if dep_libarchive.found()
@@ -51,6 +52,7 @@
       dep_libarchive,
     ],
     build_by_default: false,
+    name_suffix : 'lib',
   )
 
   foreach kmd : freedreno_kmds
diff -urN mesa-25.0.2-orig/src/freedreno/drm/meson.build mesa-25.0.2/src/freedreno/drm/meson.build
--- mesa-25.0.2-orig/src/freedreno/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/freedreno/drm/meson.build	2025-03-31 13:58:23.676309000 +0800
@@ -75,5 +75,5 @@
   # TODO
   link_with : [libvdrm],
   build_by_default : false,
+  name_suffix : 'lib',  
 )
-
diff -urN mesa-25.0.2-orig/src/freedreno/ds/meson.build mesa-25.0.2/src/freedreno/ds/meson.build
--- mesa-25.0.2-orig/src/freedreno/ds/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/freedreno/ds/meson.build	2025-03-31 13:58:23.680297100 +0800
@@ -20,6 +20,7 @@
     dep_perfetto,
     dep_valgrind,
   ],
+  name_suffix : 'lib',
 )
 
 pps_freedreno_dep = declare_dependency(
diff -urN mesa-25.0.2-orig/src/freedreno/fdl/meson.build mesa-25.0.2/src/freedreno/fdl/meson.build
--- mesa-25.0.2-orig/src/freedreno/fdl/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/freedreno/fdl/meson.build	2025-03-31 13:58:23.682957300 +0800
@@ -18,6 +18,7 @@
   gnu_symbol_visibility : 'hidden',
   dependencies : [idep_nir_headers, idep_mesautil, idep_libfreedreno_common],
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 foreach major : ['5', '6']
diff -urN mesa-25.0.2-orig/src/freedreno/ir2/meson.build mesa-25.0.2/src/freedreno/ir2/meson.build
--- mesa-25.0.2-orig/src/freedreno/ir2/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/freedreno/ir2/meson.build	2025-03-31 13:58:23.686093300 +0800
@@ -17,4 +17,5 @@
   gnu_symbol_visibility : 'hidden',
   dependencies : [],
   build_by_default : false,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/freedreno/ir3/meson.build mesa-25.0.2/src/freedreno/ir3/meson.build
--- mesa-25.0.2-orig/src/freedreno/ir3/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/freedreno/ir3/meson.build	2025-03-31 13:58:23.689083900 +0800
@@ -139,6 +139,7 @@
   link_with: [libir3encode, libfreedreno_common],
   dependencies : [idep_libir3decode, idep_nir_headers, dep_dl, idep_mesautil],
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 test('ir3_disasm',
diff -urN mesa-25.0.2-orig/src/freedreno/isa/meson.build mesa-25.0.2/src/freedreno/isa/meson.build
--- mesa-25.0.2-orig/src/freedreno/isa/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/freedreno/isa/meson.build	2025-03-31 13:58:23.692055900 +0800
@@ -40,6 +40,7 @@
     inc_src,
   ],
   gnu_symbol_visibility: 'hidden',
+  name_suffix : 'lib',
 )
 
 idep_libir3decode = declare_dependency(
@@ -90,4 +91,5 @@
     inc_freedreno,
   ],
   gnu_symbol_visibility: 'hidden',
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/freedreno/perfcntrs/meson.build mesa-25.0.2/src/freedreno/perfcntrs/meson.build
--- mesa-25.0.2-orig/src/freedreno/perfcntrs/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/freedreno/perfcntrs/meson.build	2025-03-31 13:58:23.695095900 +0800
@@ -21,6 +21,7 @@
   link_with : [libfreedreno_common],
   dependencies : idep_nir_headers,
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 dep_libconfig = dependency('libconfig', required : false)
diff -urN mesa-25.0.2-orig/src/freedreno/rnn/meson.build mesa-25.0.2/src/freedreno/rnn/meson.build
--- mesa-25.0.2-orig/src/freedreno/rnn/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/freedreno/rnn/meson.build	2025-03-31 13:58:23.699123300 +0800
@@ -26,4 +26,5 @@
   gnu_symbol_visibility: 'hidden',
   dependencies: [ dep_libxml2, idep_mesautil ],
   build_by_default: false,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/auxiliary/meson.build mesa-25.0.2/src/gallium/auxiliary/meson.build
--- mesa-25.0.2-orig/src/gallium/auxiliary/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/auxiliary/meson.build	2025-03-31 13:58:23.702080000 +0800
@@ -524,7 +524,8 @@
     dep_libdrm, dep_llvm, dep_dl, dep_m, dep_thread, dep_lmsensors, dep_ws2_32,
     idep_nir, idep_nir_headers, idep_mesautil,
   ],
-  build_by_default : false
+  build_by_default : false,
+  name_suffix : 'lib',
 )
 
 if with_tests
@@ -550,6 +551,7 @@
   include_directories: [inc_gallium, inc_include, inc_src],
   dependencies : idep_mesautil,
   build_by_default : false,
+  name_suffix : 'lib', 
 )
 
 libgalliumvl = static_library(
@@ -561,6 +563,7 @@
   include_directories : [inc_gallium, inc_include, inc_src],
   dependencies : [idep_nir, idep_mesautil],
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 # some drivers export their screen creation function globally, so all frontends have to contain the
@@ -579,4 +582,5 @@
   dependencies : [dep_libdrm, vlwinsys_deps, idep_mesautil],
   link_with : [libloader_x11],
   build_by_default : false,
+  name_suffix : 'lib', 
 )
diff -urN mesa-25.0.2-orig/src/gallium/auxiliary/pipe-loader/meson.build mesa-25.0.2/src/gallium/auxiliary/pipe-loader/meson.build
--- mesa-25.0.2-orig/src/gallium/auxiliary/pipe-loader/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/auxiliary/pipe-loader/meson.build	2025-03-31 13:58:23.705086100 +0800
@@ -52,6 +52,7 @@
   link_with : [libpipe_loader_links],
   dependencies : [dep_libdrm, idep_xmlconfig, idep_mesautil],
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 libpipe_loader_dynamic = static_library(
@@ -71,4 +72,5 @@
   link_with : [libpipe_loader_links],
   dependencies : [dep_libdrm, idep_xmlconfig, idep_mesautil],
   build_by_default : false,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/drivers/asahi/meson.build mesa-25.0.2/src/gallium/drivers/asahi/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/asahi/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/asahi/meson.build	2025-03-31 13:58:23.709117400 +0800
@@ -24,6 +24,7 @@
   c_args : [c_msvc_compat_args, '-Wno-c2x-extensions'],
   gnu_symbol_visibility : 'hidden',
   dependencies : [idep_nir, idep_mesautil, idep_agx_pack, dep_libdrm, idep_mesaclc, idep_libagx, idep_libagx_shaders_h],
+  name_suffix : 'lib',
 )
 
 driver_asahi = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/crocus/meson.build mesa-25.0.2/src/gallium/drivers/crocus/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/crocus/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/crocus/meson.build	2025-03-31 13:58:23.713108200 +0800
@@ -48,6 +48,7 @@
     ],
     gnu_symbol_visibility : 'hidden',
     dependencies : [dep_libdrm, dep_valgrind, idep_mesautil, idep_genxml, idep_nir_headers, idep_intel_dev, idep_intel_blorp_elk],
+    name_suffix : 'lib',
   )
 endforeach
 
@@ -73,6 +74,7 @@
     crocus_per_hw_ver_libs, libisl,
     libintel_perf
   ],
+  name_suffix : 'lib',
 )
 
 driver_crocus = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/d3d12/meson.build mesa-25.0.2/src/gallium/drivers/d3d12/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/d3d12/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/d3d12/meson.build	2025-03-31 13:58:23.717104900 +0800
@@ -144,6 +144,7 @@
   dependencies: [idep_nir_headers, idep_libdxil_compiler, dep_dxheaders, idep_mesautil],
   c_args: libd3d12_compiler_args,
   cpp_args: libd3d12_compiler_args,
+  name_suffix : 'lib',
 )
 
 driver_d3d12 = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/etnaviv/meson.build mesa-25.0.2/src/gallium/drivers/etnaviv/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/etnaviv/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/etnaviv/meson.build	2025-03-31 13:58:23.720120100 +0800
@@ -96,6 +96,7 @@
   ],
   link_with: [libetnaviv_drm, libetnaviv_encode],
   dependencies : [dep_libdrm, idep_nir_headers, idep_mesautil, idep_libetnaviv_decode, idep_etna_common],
+  name_suffix : 'lib',
 )
 
 driver_etnaviv = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/freedreno/meson.build mesa-25.0.2/src/gallium/drivers/freedreno/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/freedreno/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/freedreno/meson.build	2025-03-31 13:58:23.723088400 +0800
@@ -278,6 +278,7 @@
   cpp_args : [freedreno_cpp_args],
   gnu_symbol_visibility : 'hidden',
   dependencies : libfreedreno_dependencies,
+  name_suffix : 'lib',
 )
 
 driver_freedreno = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/i915/meson.build mesa-25.0.2/src/gallium/drivers/i915/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/i915/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/i915/meson.build	2025-03-31 13:58:23.726479400 +0800
@@ -51,6 +51,7 @@
   gnu_symbol_visibility : 'hidden',
   include_directories : [inc_include, inc_src, inc_gallium, inc_gallium_aux],
   dependencies : [idep_nir, idep_mesautil],
+  name_suffix : 'lib',
 )
 
 driver_i915 = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/iris/meson.build mesa-25.0.2/src/gallium/drivers/iris/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/iris/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/iris/meson.build	2025-03-31 13:58:23.730443800 +0800
@@ -72,6 +72,7 @@
     dependencies : [dep_libdrm, dep_valgrind, idep_genxml, idep_nir_headers,
                     idep_intel_driver_ds_headers, idep_intel_shaders,
                     idep_intel_blorp, idep_mesautil],
+    name_suffix : 'lib',
   )
 endforeach
 
@@ -97,6 +98,7 @@
     iris_per_hw_ver_libs, libisl,
     libintel_perf
   ],
+  name_suffix : 'lib',
 )
 
 driver_iris = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/lima/meson.build mesa-25.0.2/src/gallium/drivers/lima/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/lima/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/lima/meson.build	2025-03-31 13:58:23.734502500 +0800
@@ -87,6 +87,7 @@
     inc_panfrost
   ],
   dependencies : [dep_libdrm, idep_nir_headers, idep_mesautil],
+  name_suffix : 'lib',
 )
 
 driver_lima = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/llvmpipe/meson.build mesa-25.0.2/src/gallium/drivers/llvmpipe/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/llvmpipe/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/llvmpipe/meson.build	2025-03-31 13:58:23.736866200 +0800
@@ -103,6 +103,7 @@
   gnu_symbol_visibility : 'hidden',
   include_directories : [inc_gallium, inc_gallium_aux, inc_include, inc_src],
   dependencies : [ dep_llvm, idep_nir_headers, idep_mesautil, dep_libdrm],
+  name_suffix : 'lib',
 )
 
 driver_llvmpipe = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/nouveau/meson.build mesa-25.0.2/src/gallium/drivers/nouveau/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/nouveau/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/nouveau/meson.build	2025-03-31 13:58:23.740885800 +0800
@@ -177,6 +177,7 @@
     idep_nouveau_codegen,
     idep_nvidia_headers,
   ],
+  name_suffix : 'lib',
 )
 
 driver_nouveau = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/panfrost/meson.build mesa-25.0.2/src/gallium/drivers/panfrost/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/panfrost/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/panfrost/meson.build	2025-03-31 13:58:23.743507200 +0800
@@ -61,6 +61,7 @@
     c_args : ['-DPAN_ARCH=' + ver],
     gnu_symbol_visibility : 'hidden',
     dependencies : [idep_pan_packers, idep_nir, dep_libdrm, idep_mesautil],
+    name_suffix : 'lib',   
 )
 endforeach
 
@@ -78,6 +79,7 @@
   c_args : [c_msvc_compat_args, compile_args_panfrost],
   gnu_symbol_visibility : 'hidden',
   link_with: [libpanfrost_versions],
+  name_suffix : 'lib',  
 )
 
 driver_panfrost = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/r300/meson.build mesa-25.0.2/src/gallium/drivers/r300/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/r300/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/r300/meson.build	2025-03-31 13:58:23.747497100 +0800
@@ -120,6 +120,7 @@
   ],
   gnu_symbol_visibility : 'hidden',
   dependencies : [dep_llvm, idep_mesautil, idep_nir, idep_xmlconfig],
+  name_suffix : 'lib',
 )
 
 driver_r300 = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/r600/meson.build mesa-25.0.2/src/gallium/drivers/r600/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/r600/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/r600/meson.build	2025-03-31 13:58:23.750887600 +0800
@@ -169,6 +169,7 @@
     inc_gallium_drivers,
   ],
   dependencies: [dep_elf, dep_llvm, idep_nir, idep_nir_headers, idep_mesautil],
+  name_suffix : 'lib',
 )
 
 driver_r600 = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/r600/sfn/tests/meson.build mesa-25.0.2/src/gallium/drivers/r600/sfn/tests/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/r600/sfn/tests/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/r600/sfn/tests/meson.build	2025-03-31 13:58:23.755748300 +0800
@@ -6,7 +6,7 @@
                            inc_gallium_drivers, ],
    dependencies : [idep_gtest, dep_thread, dep_llvm, idep_nir,
                      idep_nir_headers, idep_mesautil],
-
+   name_suffix : 'lib',
 )
 
 r600_test_dep = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/radeonsi/meson.build mesa-25.0.2/src/gallium/drivers/radeonsi/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/radeonsi/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/radeonsi/meson.build	2025-03-31 13:58:23.760561200 +0800
@@ -138,6 +138,7 @@
     cpp_args : ['-DGFX_VER=@0@'.format(ver)],
     gnu_symbol_visibility : 'hidden',
     dependencies : radeonsi_deps,
+    name_suffix : 'lib',
   )
 endforeach
 
@@ -153,6 +154,7 @@
   include_directories : radeonsi_include_dirs,
   gnu_symbol_visibility : 'hidden',
   dependencies : radeonsi_deps,
+  name_suffix : 'lib',
 )
 
 driver_radeonsi = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/softpipe/meson.build mesa-25.0.2/src/gallium/drivers/softpipe/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/softpipe/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/softpipe/meson.build	2025-03-31 13:58:23.763528000 +0800
@@ -65,6 +65,7 @@
   c_args : [c_msvc_compat_args],
   gnu_symbol_visibility : 'hidden',
   dependencies : [idep_nir, idep_mesautil],
+  name_suffix : 'lib',
 )
 
 driver_softpipe = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/svga/meson.build mesa-25.0.2/src/gallium/drivers/svga/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/svga/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/svga/meson.build	2025-03-31 13:58:23.766281000 +0800
@@ -77,6 +77,7 @@
   ],
   link_with : [libglsl],
   dependencies : [idep_mesautil, idep_nir],
+  name_suffix : 'lib',
 )
 
 svga_deps = [libsvga]
diff -urN mesa-25.0.2-orig/src/gallium/drivers/tegra/meson.build mesa-25.0.2/src/gallium/drivers/tegra/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/tegra/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/tegra/meson.build	2025-03-31 13:58:23.770534900 +0800
@@ -17,6 +17,7 @@
     inc_gallium_winsys,
   ],
   dependencies : [dep_libdrm, idep_mesautil],
+  name_suffix : 'lib',
 )
 
 driver_tegra = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/v3d/meson.build mesa-25.0.2/src/gallium/drivers/v3d/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/v3d/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/v3d/meson.build	2025-03-31 13:58:23.772529500 +0800
@@ -51,6 +51,7 @@
     c_args : [v3d_args, '-DV3D_VERSION=' + ver],
     gnu_symbol_visibility : 'hidden',
     dependencies : [dep_v3d_hw, dep_libdrm, dep_valgrind, idep_nir_headers, idep_mesautil],
+    name_suffix : 'lib',
 )
 
 endforeach
@@ -73,6 +74,7 @@
     idep_nir_headers, idep_mesautil,
   ],
   link_with: [per_version_libs],
+  name_suffix : 'lib',
 )
 
 driver_v3d = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/vc4/meson.build mesa-25.0.2/src/gallium/drivers/vc4/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/vc4/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/vc4/meson.build	2025-03-31 13:58:23.775806100 +0800
@@ -81,6 +81,7 @@
     ],
     c_args : ['-mfpu=neon', '-DV3D_BUILD_NEON'],
     dependencies : vc4_deps,
+    name_suffix : 'lib',
   )
   vc4_c_args += '-DUSE_ARM_ASM'
 endif
@@ -101,6 +102,7 @@
   gnu_symbol_visibility : 'hidden',
   dependencies : vc4_deps,
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 driver_vc4 = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/drivers/virgl/meson.build mesa-25.0.2/src/gallium/drivers/virgl/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/virgl/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/virgl/meson.build	2025-03-31 13:58:23.778798100 +0800
@@ -22,6 +22,7 @@
   gnu_symbol_visibility : 'hidden',
   include_directories : [inc_include, inc_src, inc_mapi, inc_mesa, inc_gallium, inc_gallium_aux, inc_virtio],
   dependencies : [dep_libdrm, idep_mesautil, idep_xmlconfig, idep_nir],
+  name_suffix : 'lib',
 )
 
 virgl_deps = [libvirgl]
diff -urN mesa-25.0.2-orig/src/gallium/drivers/zink/meson.build mesa-25.0.2/src/gallium/drivers/zink/meson.build
--- mesa-25.0.2-orig/src/gallium/drivers/zink/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/drivers/zink/meson.build	2025-03-31 13:58:23.782814300 +0800
@@ -98,6 +98,7 @@
     idep_vulkan_wsi_defines, idep_vulkan_util, dep_libdrm, vulkan_wsi_deps
   ],
   c_args: zink_c_args,
+  name_suffix : 'lib',
 )
 
 driver_zink = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/frontends/clover/meson.build mesa-25.0.2/src/gallium/frontends/clover/meson.build
--- mesa-25.0.2-orig/src/gallium/frontends/clover/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/frontends/clover/meson.build	2025-03-31 13:58:23.786377700 +0800
@@ -50,6 +50,7 @@
   ],
   gnu_symbol_visibility : 'hidden',
   dependencies : [dep_llvm, dep_elf, dep_llvmspirvlib, idep_mesautil],
+  name_suffix : 'lib',
 )
 
 clover_files = files(
@@ -123,4 +124,5 @@
   gnu_symbol_visibility : 'hidden',
   link_with : [libclllvm],
   dependencies : [idep_mesautil, idep_nir],
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/frontends/d3d10umd/meson.build mesa-25.0.2/src/gallium/frontends/d3d10umd/meson.build
--- mesa-25.0.2-orig/src/gallium/frontends/d3d10umd/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/frontends/d3d10umd/meson.build	2025-03-31 13:58:23.790604100 +0800
@@ -30,4 +30,5 @@
   include_directories : [
     inc_include, inc_src, inc_gallium, inc_gallium_aux, inc_winddk,
   ],
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/frontends/dri/meson.build mesa-25.0.2/src/gallium/frontends/dri/meson.build
--- mesa-25.0.2-orig/src/gallium/frontends/dri/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/frontends/dri/meson.build	2025-03-31 13:58:23.793595700 +0800
@@ -64,6 +64,7 @@
     deps_for_libdri,
   ],
   link_with : [libloader],
+  name_suffix : 'lib',
 )
 
 if with_dri
diff -urN mesa-25.0.2-orig/src/gallium/frontends/glx/xlib/meson.build mesa-25.0.2/src/gallium/frontends/glx/xlib/meson.build
--- mesa-25.0.2-orig/src/gallium/frontends/glx/xlib/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/frontends/glx/xlib/meson.build	2025-03-31 13:58:23.796587900 +0800
@@ -7,4 +7,5 @@
   gnu_symbol_visibility : 'hidden',
   include_directories : [inc_include, inc_src, inc_gallium, inc_gallium_aux, inc_mapi, inc_mesa],
   dependencies : [dep_x11, dep_xext, dep_xcb, dep_glproto, idep_mesautil],
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/frontends/hgl/meson.build mesa-25.0.2/src/gallium/frontends/hgl/meson.build
--- mesa-25.0.2-orig/src/gallium/frontends/hgl/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/frontends/hgl/meson.build	2025-03-31 13:58:23.799579700 +0800
@@ -14,4 +14,5 @@
     inc_gallium, inc_gallium_aux
   ],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/frontends/lavapipe/meson.build mesa-25.0.2/src/gallium/frontends/lavapipe/meson.build
--- mesa-25.0.2-orig/src/gallium/frontends/lavapipe/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/frontends/lavapipe/meson.build	2025-03-31 13:58:23.803568900 +0800
@@ -59,5 +59,6 @@
   gnu_symbol_visibility : 'hidden',
   include_directories : [ inc_include, inc_src, inc_util, inc_gallium, inc_gallium_aux, inc_llvmpipe ],
   dependencies : [ dep_llvm, idep_nir, idep_mesautil, idep_vulkan_util, idep_vulkan_wsi,
-                   idep_vulkan_runtime, lvp_deps ]
+                   idep_vulkan_runtime, lvp_deps ],
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/frontends/nine/meson.build mesa-25.0.2/src/gallium/frontends/nine/meson.build
--- mesa-25.0.2-orig/src/gallium/frontends/nine/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/frontends/nine/meson.build	2025-03-31 13:58:23.808924200 +0800
@@ -55,5 +55,6 @@
   ],
   link_with : [
     libmesa
-  ]
+  ],
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/frontends/osmesa/meson.build mesa-25.0.2/src/gallium/frontends/osmesa/meson.build
--- mesa-25.0.2-orig/src/gallium/frontends/osmesa/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/frontends/osmesa/meson.build	2025-03-31 13:58:23.811155700 +0800
@@ -16,4 +16,5 @@
     inc_include, inc_src, inc_gallium, inc_gallium_aux, inc_mapi, inc_mesa,
   ],
   dependencies : [idep_mesautil],
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/frontends/rusticl/meson.build mesa-25.0.2/src/gallium/frontends/rusticl/meson.build
--- mesa-25.0.2-orig/src/gallium/frontends/rusticl/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/frontends/rusticl/meson.build	2025-03-31 13:58:23.814147900 +0800
@@ -178,6 +178,7 @@
   rust_args : [
     rusticl_gen_args,
   ],
+  name_suffix : 'lib',
 )
 
 rusticl_llvm_bindings_rs = rust.bindgen(
@@ -213,6 +214,7 @@
   rust_args : [
     rusticl_gen_args,
   ],
+  name_suffix : 'lib',
 )
 
 rusticl_libc_bindings_rs = rust.bindgen(
@@ -353,6 +355,7 @@
     idep_nir_headers,
     dep_valgrind,
   ],
+  name_suffix : 'lib',
 )
 
 idep_rusticl_gen = declare_dependency(
@@ -375,6 +378,7 @@
   rust_args : [
     rusticl_gen_args,
   ],
+  name_suffix : 'lib',
 )
 
 libc_rust_gen = static_library(
@@ -385,6 +389,7 @@
   rust_args : [
     rusticl_gen_args,
   ],
+  name_suffix : 'lib',
 )
 
 libmesa_rust_util = static_library(
@@ -395,6 +400,7 @@
   rust_args : [
     rusticl_args,
   ],
+  name_suffix : 'lib',
 )
 
 libmesa_rust = static_library(
@@ -410,7 +416,8 @@
     libmesa_rust_gen,
     libmesa_rust_util,
     rusticl_c,
-  ]
+  ],
+  name_suffix : 'lib',
 )
 
 rusticl_proc_macros = rust.proc_macro(
@@ -441,4 +448,5 @@
   dependencies : [
     idep_rusticl_gen,
   ],
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/frontends/teflon/meson.build mesa-25.0.2/src/gallium/frontends/teflon/meson.build
--- mesa-25.0.2-orig/src/gallium/frontends/teflon/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/frontends/teflon/meson.build	2025-03-31 13:58:23.818503600 +0800
@@ -7,5 +7,6 @@
   c_args : [  ],
   gnu_symbol_visibility : 'hidden',
   include_directories : [ inc_include, inc_src, inc_util, inc_gallium, inc_gallium_aux ],
-  dependencies : [ idep_mesautil ]
+  dependencies : [ idep_mesautil ],
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/frontends/va/meson.build mesa-25.0.2/src/gallium/frontends/va/meson.build
--- mesa-25.0.2-orig/src/gallium/frontends/va/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/frontends/va/meson.build	2025-03-31 13:58:23.822468500 +0800
@@ -23,4 +23,5 @@
     dep_va_headers, dep_x11_xcb, dep_xcb, dep_xcb_dri2, dep_xcb_dri3,
     dep_libdrm, idep_mesautil,
   ],
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/frontends/vdpau/meson.build mesa-25.0.2/src/gallium/frontends/vdpau/meson.build
--- mesa-25.0.2-orig/src/gallium/frontends/vdpau/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/frontends/vdpau/meson.build	2025-03-31 13:58:23.825460500 +0800
@@ -19,4 +19,5 @@
     inc_include, inc_src, inc_util, inc_gallium, inc_gallium_aux,
   ],
   dependencies : [dep_vdpau, dep_xcb, dep_x11_xcb, dep_xcb_dri2, dep_libdrm, idep_mesautil],
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/frontends/wgl/meson.build mesa-25.0.2/src/gallium/frontends/wgl/meson.build
--- mesa-25.0.2-orig/src/gallium/frontends/wgl/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/frontends/wgl/meson.build	2025-03-31 13:58:23.827455100 +0800
@@ -48,4 +48,5 @@
     inc_include, inc_src, inc_gallium, inc_gallium_aux, inc_mapi, inc_mesa,
   ],
   dependencies : _wgl_deps,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/frontends/xa/meson.build mesa-25.0.2/src/gallium/frontends/xa/meson.build
--- mesa-25.0.2-orig/src/gallium/frontends/xa/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/frontends/xa/meson.build	2025-03-31 13:58:23.831444400 +0800
@@ -25,6 +25,7 @@
   gnu_symbol_visibility : 'hidden',
   dependencies : idep_mesautil,
   include_directories : [inc_include, inc_src, inc_mapi, inc_mesa, inc_gallium, inc_gallium_aux],
+  name_suffix : 'lib',
 )
 
 install_headers('xa_composite.h', 'xa_context.h')
diff -urN mesa-25.0.2-orig/src/gallium/winsys/amdgpu/drm/meson.build mesa-25.0.2/src/gallium/winsys/amdgpu/drm/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/amdgpu/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/amdgpu/drm/meson.build	2025-03-31 13:58:23.833922900 +0800
@@ -20,5 +20,6 @@
   gnu_symbol_visibility : 'hidden',
   link_with : libamdgpu_addrlib,
   dependencies : [dep_llvm, dep_libdrm_amdgpu, idep_amdgfxregs_h, idep_mesautil],
-  c_args : c_args
+  c_args : c_args,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/asahi/drm/meson.build mesa-25.0.2/src/gallium/winsys/asahi/drm/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/asahi/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/asahi/drm/meson.build	2025-03-31 13:58:23.838114000 +0800
@@ -10,4 +10,5 @@
   ],
   gnu_symbol_visibility : 'hidden',
   dependencies: [dep_libdrm, idep_mesautil],
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/crocus/drm/meson.build mesa-25.0.2/src/gallium/winsys/crocus/drm/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/crocus/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/crocus/drm/meson.build	2025-03-31 13:58:23.841132800 +0800
@@ -10,4 +10,5 @@
   ],
   gnu_symbol_visibility : 'hidden',
   dependencies : [dep_valgrind, idep_intel_dev, idep_mesautil],
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/d3d12/wgl/meson.build mesa-25.0.2/src/gallium/winsys/d3d12/wgl/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/d3d12/wgl/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/d3d12/wgl/meson.build	2025-03-31 13:58:23.843777200 +0800
@@ -14,6 +14,7 @@
   include_directories : [inc_src, inc_wgl, inc_include, inc_gallium, inc_gallium_aux, inc_gallium_drivers],
   dependencies : [dep_dxheaders, idep_nir_headers, idep_libdxil_compiler, idep_mesautil],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',
 )
 
 winsys_d3d12_wgl = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/winsys/etnaviv/drm/meson.build mesa-25.0.2/src/gallium/winsys/etnaviv/drm/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/etnaviv/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/etnaviv/drm/meson.build	2025-03-31 13:58:23.846771900 +0800
@@ -10,4 +10,5 @@
   ],
   link_with: libetnaviv_drm,
   dependencies : [dep_libdrm, idep_nir_headers, idep_mesautil, idep_etna_common, idep_mesautil],
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/freedreno/drm/meson.build mesa-25.0.2/src/gallium/winsys/freedreno/drm/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/freedreno/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/freedreno/drm/meson.build	2025-03-31 13:58:23.849787200 +0800
@@ -16,4 +16,5 @@
     dep_libdrm,
     idep_mesautil,
   ],
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/i915/drm/meson.build mesa-25.0.2/src/gallium/winsys/i915/drm/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/i915/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/i915/drm/meson.build	2025-03-31 13:58:23.851782000 +0800
@@ -12,4 +12,5 @@
   ],
   link_with : [libintel_common],
   dependencies : [dep_libdrm, dep_libdrm_intel, idep_intel_dev_wa, idep_mesautil],
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/iris/drm/meson.build mesa-25.0.2/src/gallium/winsys/iris/drm/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/iris/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/iris/drm/meson.build	2025-03-31 13:58:23.854750200 +0800
@@ -10,4 +10,5 @@
   ],
   gnu_symbol_visibility : 'hidden',
   dependencies: idep_mesautil,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/kmsro/drm/meson.build mesa-25.0.2/src/gallium/winsys/kmsro/drm/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/kmsro/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/kmsro/drm/meson.build	2025-03-31 13:58:23.857741500 +0800
@@ -12,6 +12,7 @@
   c_args : [renderonly_drivers_c_args],
   gnu_symbol_visibility : 'hidden',
   dependencies: [dep_libdrm, idep_mesautil],
+  name_suffix : 'lib',
 )
 
 driver_kmsro = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/winsys/lima/drm/meson.build mesa-25.0.2/src/gallium/winsys/lima/drm/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/lima/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/lima/drm/meson.build	2025-03-31 13:58:23.859736900 +0800
@@ -9,4 +9,5 @@
     inc_gallium, inc_gallium_aux, inc_gallium_drivers,
   ],
   dependencies : idep_mesautil,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/nouveau/drm/meson.build mesa-25.0.2/src/gallium/winsys/nouveau/drm/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/nouveau/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/nouveau/drm/meson.build	2025-03-31 13:58:23.862731100 +0800
@@ -24,6 +24,7 @@
   ],
   gnu_symbol_visibility : 'hidden',
   dependencies : [dep_libdrm, idep_mesautil, idep_nvidia_headers],
+  name_suffix : 'lib',
 )
 
 idep_libnouveauwinsys = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/winsys/panfrost/drm/meson.build mesa-25.0.2/src/gallium/winsys/panfrost/drm/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/panfrost/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/panfrost/drm/meson.build	2025-03-31 13:58:23.865721100 +0800
@@ -10,4 +10,5 @@
   ],
   gnu_symbol_visibility : 'hidden',
   dependencies: [dep_libdrm, idep_mesautil],
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/radeon/drm/meson.build mesa-25.0.2/src/gallium/winsys/radeon/drm/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/radeon/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/radeon/drm/meson.build	2025-03-31 13:58:23.868713200 +0800
@@ -15,4 +15,5 @@
   include_directories : [inc_src, inc_include, inc_gallium, inc_gallium_aux],
   gnu_symbol_visibility : 'hidden',
   dependencies : [idep_mesautil, dep_libdrm],
+  name_suffix : 'lib', 
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/svga/drm/meson.build mesa-25.0.2/src/gallium/winsys/svga/drm/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/svga/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/svga/drm/meson.build	2025-03-31 13:58:23.871925200 +0800
@@ -27,4 +27,5 @@
     include_directories('../../../drivers/svga', '../../../drivers/svga/include'),
   ],
   dependencies : [dep_libdrm, idep_mesautil],
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/sw/dri/meson.build mesa-25.0.2/src/gallium/winsys/sw/dri/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/sw/dri/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/sw/dri/meson.build	2025-03-31 13:58:23.874361300 +0800
@@ -8,4 +8,5 @@
   gnu_symbol_visibility : 'hidden',
   include_directories : [inc_gallium, inc_include, inc_src, inc_gallium_aux],
   dependencies : idep_mesautil,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/sw/gdi/meson.build mesa-25.0.2/src/gallium/winsys/sw/gdi/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/sw/gdi/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/sw/gdi/meson.build	2025-03-31 13:58:23.876355600 +0800
@@ -8,4 +8,5 @@
     inc_src, inc_include, inc_gallium, inc_gallium_aux, inc_gallium_drivers, inc_frontends
   ],
   dependencies: idep_mesautil,
+  name_suffix : 'lib', 
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/sw/hgl/meson.build mesa-25.0.2/src/gallium/winsys/sw/hgl/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/sw/hgl/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/sw/hgl/meson.build	2025-03-31 13:58:23.879347500 +0800
@@ -10,4 +10,5 @@
   ],
   dependencies: idep_mesautil,
   build_by_default : false,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/sw/kms-dri/meson.build mesa-25.0.2/src/gallium/winsys/sw/kms-dri/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/sw/kms-dri/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/sw/kms-dri/meson.build	2025-03-31 13:58:23.882339700 +0800
@@ -7,4 +7,5 @@
   gnu_symbol_visibility : 'hidden',
   include_directories : [inc_gallium, inc_include, inc_src, inc_gallium_aux],
   dependencies : [dep_libdrm, idep_mesautil],
+  name_suffix : 'lib',  
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/sw/null/meson.build mesa-25.0.2/src/gallium/winsys/sw/null/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/sw/null/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/sw/null/meson.build	2025-03-31 13:58:23.884334400 +0800
@@ -8,4 +8,5 @@
   include_directories : [inc_gallium, inc_src, inc_gallium_aux, inc_include],
   dependencies : idep_mesautil,
   build_by_default : false,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/sw/wrapper/meson.build mesa-25.0.2/src/gallium/winsys/sw/wrapper/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/sw/wrapper/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/sw/wrapper/meson.build	2025-03-31 13:58:23.887326200 +0800
@@ -8,4 +8,5 @@
   include_directories : [inc_gallium, inc_include, inc_src, inc_gallium_aux],
   dependencies : idep_mesautil,
   build_by_default : false,
+  name_suffix : 'lib', 
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/sw/xlib/meson.build mesa-25.0.2/src/gallium/winsys/sw/xlib/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/sw/xlib/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/sw/xlib/meson.build	2025-03-31 13:58:23.889320800 +0800
@@ -7,4 +7,5 @@
   gnu_symbol_visibility : 'hidden',
   include_directories : [inc_include, inc_src, inc_mapi, inc_mesa, inc_gallium, inc_gallium_aux],
   dependencies : [dep_x11, dep_xext, dep_xcb, idep_mesautil],
+  name_suffix : 'lib', 
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/tegra/drm/meson.build mesa-25.0.2/src/gallium/winsys/tegra/drm/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/tegra/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/tegra/drm/meson.build	2025-03-31 13:58:23.892312800 +0800
@@ -9,6 +9,7 @@
     inc_gallium_winsys
   ],
   dependencies : idep_mesautil,
+  name_suffix : 'lib',
 )
 
 driver_tegra = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/winsys/v3d/drm/meson.build mesa-25.0.2/src/gallium/winsys/v3d/drm/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/v3d/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/v3d/drm/meson.build	2025-03-31 13:58:23.894307700 +0800
@@ -10,4 +10,5 @@
   ],
   gnu_symbol_visibility : 'hidden',
   dependencies : idep_mesautil,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/vc4/drm/meson.build mesa-25.0.2/src/gallium/winsys/vc4/drm/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/vc4/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/vc4/drm/meson.build	2025-03-31 13:58:23.897299700 +0800
@@ -22,4 +22,5 @@
   ],
   c_args : [vc4_winsys_c_args],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib', 
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/virgl/common/meson.build mesa-25.0.2/src/gallium/winsys/virgl/common/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/virgl/common/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/virgl/common/meson.build	2025-03-31 13:58:23.900291500 +0800
@@ -7,6 +7,7 @@
   include_directories : [inc_include, inc_src, inc_mapi, inc_mesa, inc_gallium, inc_gallium_aux, inc_gallium_drivers],
   gnu_symbol_visibility : 'hidden',
   dependencies: idep_mesautil,
+  name_suffix : 'lib', 
 )
 
 dep_libvirglcommon = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gallium/winsys/virgl/drm/meson.build mesa-25.0.2/src/gallium/winsys/virgl/drm/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/virgl/drm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/virgl/drm/meson.build	2025-03-31 13:58:23.902286400 +0800
@@ -7,4 +7,5 @@
   include_directories : [inc_include, inc_src, inc_mapi, inc_mesa, inc_gallium, inc_gallium_aux, inc_gallium_drivers, inc_virtio],
   dependencies : [dep_libdrm, dep_libvirglcommon, idep_mesautil, idep_nir_headers],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib', 
 )
diff -urN mesa-25.0.2-orig/src/gallium/winsys/virgl/vtest/meson.build mesa-25.0.2/src/gallium/winsys/virgl/vtest/meson.build
--- mesa-25.0.2-orig/src/gallium/winsys/virgl/vtest/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gallium/winsys/virgl/vtest/meson.build	2025-03-31 13:58:23.905451700 +0800
@@ -7,4 +7,5 @@
   include_directories : [inc_include, inc_src, inc_mapi, inc_mesa, inc_gallium, inc_gallium_aux, inc_gallium_drivers, inc_virtio],
   dependencies : [dep_libvirglcommon, idep_mesautil],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/getopt/meson.build mesa-25.0.2/src/getopt/meson.build
--- mesa-25.0.2-orig/src/getopt/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/getopt/meson.build	2025-03-31 13:58:23.907448500 +0800
@@ -4,6 +4,7 @@
 libgetopt = static_library(
   'getopt',
   'getopt_long.c',
+   name_suffix : 'lib',
 )
 
 idep_getopt = declare_dependency(
diff -urN mesa-25.0.2-orig/src/gfxstream/aemu/meson.build mesa-25.0.2/src/gfxstream/aemu/meson.build
--- mesa-25.0.2-orig/src/gfxstream/aemu/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gfxstream/aemu/meson.build	2025-03-31 13:58:23.910440800 +0800
@@ -12,4 +12,5 @@
    'aemu',
    files_libaemu,
    include_directories: [inc_aemu],
+   name_suffix : 'lib', 
 )
diff -urN mesa-25.0.2-orig/src/gfxstream/guest/connection-manager/meson.build mesa-25.0.2/src/gfxstream/guest/connection-manager/meson.build
--- mesa-25.0.2-orig/src/gfxstream/guest/connection-manager/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gfxstream/guest/connection-manager/meson.build	2025-03-31 13:58:23.915427500 +0800
@@ -18,5 +18,6 @@
                          inc_guest_iostream, inc_goldfish_address_space,
                          inc_platform_virtgpu, inc_src, inc_aemu],
    link_with: [lib_goldfish_address_space, libplatform_virtgpu],
-   dependencies: dep_libdrm
+   dependencies: dep_libdrm,
+   name_suffix : 'lib',  
 )
diff -urN mesa-25.0.2-orig/src/gfxstream/guest/GoldfishAddressSpace/meson.build mesa-25.0.2/src/gfxstream/guest/GoldfishAddressSpace/meson.build
--- mesa-25.0.2-orig/src/gfxstream/guest/GoldfishAddressSpace/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gfxstream/guest/GoldfishAddressSpace/meson.build	2025-03-31 13:58:23.919735100 +0800
@@ -24,5 +24,6 @@
                          inc_aemu,
                          inc_src],
    link_with: [libplatform_virtgpu],
-   dependencies: dep_libdrm
+   dependencies: dep_libdrm,
+   name_suffix : 'lib',  
 )
diff -urN mesa-25.0.2-orig/src/gfxstream/guest/platform/kumquat/meson.build mesa-25.0.2/src/gfxstream/guest/platform/kumquat/meson.build
--- mesa-25.0.2-orig/src/gfxstream/guest/platform/kumquat/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gfxstream/guest/platform/kumquat/meson.build	2025-03-31 13:58:23.924721000 +0800
@@ -18,4 +18,5 @@
    cpp_args: gfxstream_guest_args,
    include_directories: [inc_platform_virtgpu, inc_src],
    dependencies: virtgpu_kumquat_dep,
+   name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gfxstream/guest/platform/linux/meson.build mesa-25.0.2/src/gfxstream/guest/platform/linux/meson.build
--- mesa-25.0.2-orig/src/gfxstream/guest/platform/linux/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gfxstream/guest/platform/linux/meson.build	2025-03-31 13:58:23.927712900 +0800
@@ -14,4 +14,5 @@
    cpp_args: gfxstream_guest_args,
    include_directories: [inc_platform_virtgpu, inc_src, inc_include],
    dependencies: dep_libdrm,
+   name_suffix : 'lib',  
 )
diff -urN mesa-25.0.2-orig/src/gfxstream/guest/platform/meson.build mesa-25.0.2/src/gfxstream/guest/platform/meson.build
--- mesa-25.0.2-orig/src/gfxstream/guest/platform/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gfxstream/guest/platform/meson.build	2025-03-31 13:58:23.932700200 +0800
@@ -24,5 +24,6 @@
    files_libplatform_virtgpu,
    cpp_args: gfxstream_guest_args,
    include_directories: [inc_platform_virtgpu, inc_src],
-   link_with: libplatform_virtgpu
+   link_with: libplatform_virtgpu,
+   name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gfxstream/guest/platform/windows/meson.build mesa-25.0.2/src/gfxstream/guest/platform/windows/meson.build
--- mesa-25.0.2-orig/src/gfxstream/guest/platform/windows/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gfxstream/guest/platform/windows/meson.build	2025-03-31 13:58:23.936713200 +0800
@@ -13,4 +13,5 @@
    files_libplatform_virtgpu_windows,
    cpp_args: gfxstream_guest_args,
    include_directories: [inc_platform_virtgpu, inc_src, inc_include],
+   name_suffix : 'lib', 
 )
diff -urN mesa-25.0.2-orig/src/gfxstream/guest/vulkan/stubs/meson.build mesa-25.0.2/src/gfxstream/guest/vulkan/stubs/meson.build
--- mesa-25.0.2-orig/src/gfxstream/guest/vulkan/stubs/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gfxstream/guest/vulkan/stubs/meson.build	2025-03-31 13:58:23.940678300 +0800
@@ -12,4 +12,5 @@
    cpp_args: gfxstream_guest_args,
    include_directories: [inc_connection_manager, inc_platform_virtgpu,
                          inc_guest_iostream],
+   name_suffix : 'lib',                      
 )
diff -urN mesa-25.0.2-orig/src/glx/apple/meson.build mesa-25.0.2/src/glx/apple/meson.build
--- mesa-25.0.2-orig/src/glx/apple/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/glx/apple/meson.build	2025-03-31 13:58:23.943670800 +0800
@@ -41,4 +41,5 @@
   dependencies: [dep_xext, dep_xplugin],
   gnu_symbol_visibility : 'hidden',
   build_by_default: false,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/glx/meson.build mesa-25.0.2/src/glx/meson.build
--- mesa-25.0.2-orig/src/glx/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/glx/meson.build	2025-03-31 13:58:23.946662900 +0800
@@ -128,6 +128,7 @@
     idep_mesautil, idep_xmlconfig,
     dep_libdrm, dep_dri2proto, dep_glproto, dep_x11, dep_xext, dep_glvnd, dep_xxf86vm, dep_xshmfence, dep_xfixes,
   ],
+  name_suffix : 'lib',
 )
 
 libgl = shared_library(
diff -urN mesa-25.0.2-orig/src/glx/windows/meson.build mesa-25.0.2/src/glx/windows/meson.build
--- mesa-25.0.2-orig/src/glx/windows/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/glx/windows/meson.build	2025-03-31 13:58:23.949655200 +0800
@@ -30,6 +30,7 @@
   [files_libwindowsdri, files_windowsdriproto],
   dependencies: dep_xext,
   build_by_default: false,
+  name_suffix : 'lib',
 )
 
 # library for native GL on windows
@@ -48,4 +49,5 @@
   include_directories: [inc_include, inc_src, inc_glapi],
   gnu_symbol_visibility : 'hidden',
   build_by_default: false,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/gtest/meson.build mesa-25.0.2/src/gtest/meson.build
--- mesa-25.0.2-orig/src/gtest/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/gtest/meson.build	2025-03-31 13:58:23.952023900 +0800
@@ -7,6 +7,7 @@
   include_directories : include_directories('include'),
   override_options: 'warning_level=1',
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 idep_gtest = declare_dependency(
diff -urN mesa-25.0.2-orig/src/imagination/common/meson.build mesa-25.0.2/src/imagination/common/meson.build
--- mesa-25.0.2-orig/src/imagination/common/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/imagination/common/meson.build	2025-03-31 13:58:23.954018400 +0800
@@ -18,4 +18,5 @@
   ],
   c_args : [imagination_c_args, no_override_init_args],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/imagination/pco/meson.build mesa-25.0.2/src/imagination/pco/meson.build
--- mesa-25.0.2-orig/src/imagination/pco/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/imagination/pco/meson.build	2025-03-31 13:58:23.956030500 +0800
@@ -107,4 +107,5 @@
   dependencies : [idep_mesautil, idep_nir, idep_pco_pygen],
   gnu_symbol_visibility : 'hidden',
   install : false,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/imagination/vulkan/pds/meson.build mesa-25.0.2/src/imagination/vulkan/pds/meson.build
--- mesa-25.0.2-orig/src/imagination/vulkan/pds/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/imagination/vulkan/pds/meson.build	2025-03-31 13:58:23.958007800 +0800
@@ -29,4 +29,5 @@
   ],
   gnu_symbol_visibility : 'hidden',
   pic : true,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/imgui/meson.build mesa-25.0.2/src/imgui/meson.build
--- mesa-25.0.2-orig/src/imgui/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/imgui/meson.build	2025-03-31 13:58:23.961001200 +0800
@@ -2,7 +2,8 @@
   'imgui_core',
   files('imgui.cpp', 'imgui_draw.cpp', 'imgui_widgets.cpp'),
   cpp_args : ['-w'],
-  install : false
+  install : false,
+  name_suffix : 'lib',
 )
 
 libimgui_core_dep = declare_dependency(
diff -urN mesa-25.0.2-orig/src/intel/blorp/meson.build mesa-25.0.2/src/intel/blorp/meson.build
--- mesa-25.0.2-orig/src/intel/blorp/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/intel/blorp/meson.build	2025-03-31 13:58:23.963991800 +0800
@@ -28,6 +28,7 @@
   gnu_symbol_visibility : 'hidden',
   dependencies : [idep_nir_headers, idep_genxml, idep_mesautil, idep_intel_dev],
   build_by_default: false,
+  name_suffix : 'lib',
 )
 
 idep_intel_blorp = declare_dependency(
@@ -43,6 +44,7 @@
     gnu_symbol_visibility : 'hidden',
     dependencies : [idep_nir_headers, idep_genxml, idep_mesautil, idep_intel_dev],
     build_by_default: true, # FIXME XXX
+    name_suffix : 'lib',
   )
 
   idep_intel_blorp_elk = declare_dependency(
diff -urN mesa-25.0.2-orig/src/intel/common/meson.build mesa-25.0.2/src/intel/common/meson.build
--- mesa-25.0.2-orig/src/intel/common/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/intel/common/meson.build	2025-03-31 13:58:23.965986800 +0800
@@ -56,6 +56,7 @@
     include_directories : [inc_include, inc_src],
     c_args : [no_override_init_args] + clflushopt_args,
     gnu_symbol_visibility : 'hidden',
+    name_suffix : 'lib',
   )
   libintel_common_links += libintel_clflushopt
 endif
@@ -68,6 +69,7 @@
   gnu_symbol_visibility : 'hidden',
   link_with : libintel_common_links,
   dependencies : [dep_libdrm, dep_thread, idep_genxml, idep_mesautil, idep_intel_dev],
+  name_suffix : 'lib',
 )
 
 libintel_extra_dependencies = []
diff -urN mesa-25.0.2-orig/src/intel/compiler/elk/meson.build mesa-25.0.2/src/intel/compiler/elk/meson.build
--- mesa-25.0.2-orig/src/intel/compiler/elk/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/intel/compiler/elk/meson.build	2025-03-31 13:58:23.969092400 +0800
@@ -134,6 +134,7 @@
   gnu_symbol_visibility : 'hidden',
   dependencies : [idep_nir_headers, idep_mesautil, idep_intel_dev],
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 idep_intel_compiler_elk = declare_dependency(
diff -urN mesa-25.0.2-orig/src/intel/compiler/meson.build mesa-25.0.2/src/intel/compiler/meson.build
--- mesa-25.0.2-orig/src/intel/compiler/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/intel/compiler/meson.build	2025-03-31 13:58:23.972247000 +0800
@@ -155,6 +155,7 @@
   gnu_symbol_visibility : 'hidden',
   dependencies : [idep_nir_headers, idep_mesautil, idep_intel_dev],
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 idep_intel_compiler_brw = declare_dependency(
@@ -273,6 +274,7 @@
   c_args : [no_override_init_args],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 idep_brw_asm = declare_dependency(
diff -urN mesa-25.0.2-orig/src/intel/decoder/meson.build mesa-25.0.2/src/intel/decoder/meson.build
--- mesa-25.0.2-orig/src/intel/decoder/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/intel/decoder/meson.build	2025-03-31 13:58:23.974912400 +0800
@@ -33,6 +33,7 @@
   gnu_symbol_visibility : 'hidden',
   dependencies : libintel_decoder_deps,
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 idep_intel_decoder = declare_dependency(
@@ -48,6 +49,7 @@
   gnu_symbol_visibility : 'hidden',
   dependencies : idep_intel_decoder,
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 idep_intel_decoder_brw = declare_dependency(
@@ -64,6 +66,7 @@
     gnu_symbol_visibility : 'hidden',
     dependencies : idep_intel_decoder,
     build_by_default : false,
+    name_suffix : 'lib',
   )
 
   idep_intel_decoder_elk = declare_dependency(
diff -urN mesa-25.0.2-orig/src/intel/dev/meson.build mesa-25.0.2/src/intel/dev/meson.build
--- mesa-25.0.2-orig/src/intel/dev/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/intel/dev/meson.build	2025-03-31 13:58:23.977904900 +0800
@@ -48,6 +48,7 @@
                   idep_intel_dev_info_gen],
   c_args : [no_override_init_args],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',
 )
 
 idep_intel_dev = declare_dependency(
diff -urN mesa-25.0.2-orig/src/intel/ds/meson.build mesa-25.0.2/src/intel/ds/meson.build
--- mesa-25.0.2-orig/src/intel/ds/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/intel/ds/meson.build	2025-03-31 13:58:23.980871200 +0800
@@ -40,6 +40,7 @@
   link_with : [libintel_perf, libintel_common],
   dependencies : [libintel_driver_ds_deps, idep_intel_dev, idep_intel_blorp],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',
 )
 
 idep_intel_driver_ds = declare_dependency(
@@ -60,6 +61,7 @@
     include_directories: [inc_tool, inc_src, inc_include, inc_intel],
     link_with: [libintel_perf, libintel_common],
     dependencies: [dep_perfetto, dep_libdrm, idep_mesautil, idep_intel_dev],
+    name_suffix : 'lib',
   )
 
   compile_args_pps_intel = ['-DPPS_INTEL']
diff -urN mesa-25.0.2-orig/src/intel/executor/meson.build mesa-25.0.2/src/intel/executor/meson.build
--- mesa-25.0.2-orig/src/intel/executor/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/intel/executor/meson.build	2025-03-31 13:58:23.983220200 +0800
@@ -31,6 +31,7 @@
       dep_valgrind,
       idep_genxml,
     ],
+    name_suffix : 'lib',
   )
 endforeach
 
diff -urN mesa-25.0.2-orig/src/intel/isl/meson.build mesa-25.0.2/src/intel/isl/meson.build
--- mesa-25.0.2-orig/src/intel/isl/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/intel/isl/meson.build	2025-03-31 13:58:23.985915500 +0800
@@ -65,6 +65,7 @@
       '-DGFX_VERx10=@0@'.format(_gfx_ver)
     ],
     gnu_symbol_visibility : 'hidden',
+    name_suffix : 'lib',
   )
 endforeach
 
@@ -93,7 +94,8 @@
   dependencies : [idep_mesautil, idep_intel_dev],
   c_args : [no_override_init_args, sse2_arg],
   gnu_symbol_visibility : 'hidden',
-  extra_files : ['isl_tiled_memcpy.c']
+  extra_files : ['isl_tiled_memcpy.c'],
+  name_suffix : 'lib',
 )
 
 if with_sse41
@@ -108,6 +110,7 @@
     c_args : [no_override_init_args, sse2_arg, sse41_args],
     gnu_symbol_visibility : 'hidden',
     extra_files : ['isl_tiled_memcpy.c']
+    name_suffix : 'lib',
   )
 else
   isl_tiled_memcpy_sse41 = []
@@ -132,6 +135,7 @@
   dependencies : [idep_mesautil, idep_intel_dev],
   c_args : [no_override_init_args],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',
 )
 
 if with_intel_tools
diff -urN mesa-25.0.2-orig/src/intel/perf/meson.build mesa-25.0.2/src/intel/perf/meson.build
--- mesa-25.0.2-orig/src/intel/perf/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/intel/perf/meson.build	2025-03-31 13:58:23.988907400 +0800
@@ -52,6 +52,7 @@
   cpp_args : [sse2_arg],
   gnu_symbol_visibility : 'hidden',
   dependencies : [idep_intel_dev],
+  name_suffix : 'lib',
 )
 
 if with_intel_tools
diff -urN mesa-25.0.2-orig/src/intel/tools/imgui/meson.build mesa-25.0.2/src/intel/tools/imgui/meson.build
--- mesa-25.0.2-orig/src/intel/tools/imgui/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/intel/tools/imgui/meson.build	2025-03-31 13:58:23.992647900 +0800
@@ -9,7 +9,8 @@
   include_directories: inc_src,
   dependencies : libintel_imgui_gtk_deps,
   cpp_args : ['-w'],
-  install: false
+  install: false,
+  name_suffix : 'lib',
 )
 
 libintel_imgui_gtk_dep = declare_dependency(
diff -urN mesa-25.0.2-orig/src/intel/tools/meson.build mesa-25.0.2/src/intel/tools/meson.build
--- mesa-25.0.2-orig/src/intel/tools/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/intel/tools/meson.build	2025-03-31 13:58:23.996639900 +0800
@@ -24,6 +24,7 @@
   c_args : [intel_common_args, no_override_init_args],
   gnu_symbol_visibility : 'hidden',
   install : false,
+  name_suffix : 'lib',
 )
 
 idep_libintel_tools = declare_dependency(
diff -urN mesa-25.0.2-orig/src/intel/vulkan/grl/meson.build mesa-25.0.2/src/intel/vulkan/grl/meson.build
--- mesa-25.0.2-orig/src/intel/vulkan/grl/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/intel/vulkan/grl/meson.build	2025-03-31 13:58:23.998633800 +0800
@@ -158,6 +158,7 @@
       idep_vulkan_runtime_headers, idep_anv_headers, idep_genxml,
     ],
     gnu_symbol_visibility : 'hidden',
+    name_suffix : 'lib',
   )
 endforeach
 
@@ -176,6 +177,7 @@
   ],
   link_whole : [grl_genX_libs],
   dependencies : [libgrl_deps, idep_anv_headers],
+  name_suffix : 'lib',
 )
 idep_grl = declare_dependency(
   link_with : libgrl,
diff -urN mesa-25.0.2-orig/src/intel/vulkan/meson.build mesa-25.0.2/src/intel/vulkan/meson.build
--- mesa-25.0.2-orig/src/intel/vulkan/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/intel/vulkan/meson.build	2025-03-31 13:58:24.002010000 +0800
@@ -138,6 +138,7 @@
       idep_intel_driver_ds_headers, idep_grl,
       idep_intel_shaders, idep_intel_blorp,
     ],
+    name_suffix : 'lib',
   )
 endforeach
 
@@ -261,7 +262,8 @@
   c_args : anv_flags,
   cpp_args : anv_cpp_flags,
   gnu_symbol_visibility : 'hidden',
-  dependencies : anv_deps
+  dependencies : anv_deps,
+  name_suffix : 'lib',
 )
 
 libvulkan_intel = shared_library(
@@ -321,6 +323,7 @@
     ],
     c_args : anv_flags,
     gnu_symbol_visibility : 'hidden',
+    name_suffix : 'lib',
   )
 
   files_anv_tests = files(
diff -urN mesa-25.0.2-orig/src/intel/vulkan_hasvk/meson.build mesa-25.0.2/src/intel/vulkan_hasvk/meson.build
--- mesa-25.0.2-orig/src/intel/vulkan_hasvk/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/intel/vulkan_hasvk/meson.build	2025-03-31 13:58:24.004004400 +0800
@@ -81,6 +81,7 @@
       idep_vulkan_runtime_headers, idep_intel_driver_ds_headers,
       idep_intel_blorp_elk, idep_mesautil,
     ],
+    name_suffix : 'lib', 
   )
 endforeach
 
@@ -166,6 +167,7 @@
   cpp_args : anv_cpp_flags,
   gnu_symbol_visibility : 'hidden',
   dependencies : anv_deps,
+  name_suffix : 'lib',
 )
 
 libvulkan_intel_hasvk = shared_library(
@@ -226,6 +228,7 @@
     ],
     c_args : anv_flags,
     gnu_symbol_visibility : 'hidden',
+    name_suffix : 'lib',
   )
 
   files_hasvk_tests = files(
diff -urN mesa-25.0.2-orig/src/loader/meson.build mesa-25.0.2/src/loader/meson.build
--- mesa-25.0.2-orig/src/loader/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/loader/meson.build	2025-03-31 13:58:24.007023600 +0800
@@ -30,4 +30,5 @@
   include_directories : [inc_include, inc_src, inc_util, inc_gallium],
   dependencies : [idep_mesautil, dep_libdrm, dep_thread, dep_xcb, dep_xcb_xrandr],
   build_by_default : false,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/mapi/glapi/meson.build mesa-25.0.2/src/mapi/glapi/meson.build
--- mesa-25.0.2-orig/src/mapi/glapi/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/mapi/glapi/meson.build	2025-03-31 14:07:03.007841300 +0800
@@ -67,6 +67,7 @@
   c_args : [c_msvc_compat_args, bridge_glapi_args],
   dependencies : [dep_thread, idep_mesautil],
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 if with_shared_glapi
@@ -79,6 +80,7 @@
     c_args : [c_msvc_compat_args, static_glapi_args],
     dependencies : [dep_thread, idep_mesautil],
     build_by_default : false,
+    name_suffix : 'lib',
   )
 endif
 
diff -urN mesa-25.0.2-orig/src/mapi/shared-glapi/meson.build mesa-25.0.2/src/mapi/shared-glapi/meson.build
--- mesa-25.0.2-orig/src/mapi/shared-glapi/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/mapi/shared-glapi/meson.build	2025-03-31 13:58:24.011597800 +0800
@@ -41,5 +41,6 @@
   include_directories : [inc_src, inc_include, inc_mapi],
   dependencies : [dep_thread, idep_mesautil],
   install : false,
+  name_suffix : 'lib',
 )
 libglapi_build_dir = meson.current_build_dir()
diff -urN mesa-25.0.2-orig/src/mesa/meson.build mesa-25.0.2/src/mesa/meson.build
--- mesa-25.0.2-orig/src/mesa/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/mesa/meson.build	2025-03-31 13:58:24.014584400 +0800
@@ -435,6 +435,7 @@
     c_args : [c_msvc_compat_args, sse41_args],
     include_directories : [inc_include, inc_src, inc_mapi, inc_mesa, inc_gallium, inc_gallium_aux],
     gnu_symbol_visibility : 'hidden',
+    name_suffix : 'lib',
   )
 else
   libmesa_sse41 = []
@@ -461,4 +462,5 @@
   link_with : [libmesa_sse41],
   dependencies : [idep_libglsl, idep_nir, idep_vtn, dep_vdpau, idep_mesautil],
   build_by_default : false,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/microsoft/compiler/meson.build mesa-25.0.2/src/microsoft/compiler/meson.build
--- mesa-25.0.2-orig/src/microsoft/compiler/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/microsoft/compiler/meson.build	2025-03-31 13:58:24.017959400 +0800
@@ -38,6 +38,7 @@
   include_directories : [inc_include, inc_src],
   dependencies: [idep_nir_headers, dep_valgrind],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',
 )
 
 idep_libdxil_compiler = declare_dependency(
diff -urN mesa-25.0.2-orig/src/nouveau/codegen/meson.build mesa-25.0.2/src/nouveau/codegen/meson.build
--- mesa-25.0.2-orig/src/nouveau/codegen/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/nouveau/codegen/meson.build	2025-03-31 13:58:24.019954400 +0800
@@ -54,6 +54,7 @@
   ],
   gnu_symbol_visibility : 'hidden',
   dependencies : [dep_libdrm, idep_mesautil, idep_nir_headers],
+  name_suffix : 'lib',
 )
 
 idep_nouveau_codegen = declare_dependency(
diff -urN mesa-25.0.2-orig/src/nouveau/compiler/meson.build mesa-25.0.2/src/nouveau/compiler/meson.build
--- mesa-25.0.2-orig/src/nouveau/compiler/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/nouveau/compiler/meson.build	2025-03-31 13:58:24.022946400 +0800
@@ -42,6 +42,7 @@
   gnu_symbol_visibility : 'hidden',
   rust_abi : 'rust',
   rust_args : nak_rust_args,
+  name_suffix : 'lib',
 )
 
 idep_acorn_rs = declare_dependency(
@@ -54,6 +55,7 @@
   gnu_symbol_visibility : 'hidden',
   rust_abi : 'rust',
   rust_args : nak_rust_args,
+  name_suffix : 'lib',
 )
 
 idep_bitview_rs = declare_dependency(
@@ -103,6 +105,7 @@
     idep_compiler_rs,
   ],
   rust_abi : 'rust',
+  name_suffix : 'lib',
 )
 
 _libnak_ir_proc_rs = rust.proc_macro(
@@ -131,6 +134,7 @@
     _libnak_bindings_rs,
     _libnak_ir_proc_rs,
   ],
+  name_suffix : 'lib',
 )
 
 # TODO: Linking Rust executables (such as unit tests) doesn't play nicely
@@ -181,6 +185,7 @@
   link_with : [_libnak_rs],
   c_args : [no_override_init_args],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',
 )
 
 idep_nak = declare_dependency(
diff -urN mesa-25.0.2-orig/src/nouveau/headers/meson.build mesa-25.0.2/src/nouveau/headers/meson.build
--- mesa-25.0.2-orig/src/nouveau/headers/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/nouveau/headers/meson.build	2025-03-31 13:58:24.025707400 +0800
@@ -52,6 +52,7 @@
   include_directories : ['.', 'nvidia/classes', inc_include, inc_src],
   c_args : [no_override_init_args],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',
 )
 
 idep_nvidia_headers = declare_dependency(
@@ -141,6 +142,7 @@
     _nvidia_headers_lib_rs,
     gnu_symbol_visibility : 'hidden',
     rust_abi : 'rust',
+    name_suffix : 'lib',
   )
 
   idep_nvidia_headers_rs = declare_dependency(
@@ -153,6 +155,7 @@
     gnu_symbol_visibility: 'hidden',
     rust_abi: 'rust',
     dependencies: [idep_nvidia_headers_rs],
+    name_suffix : 'lib',
   )
 
   idep_nv_push_rs = declare_dependency(
diff -urN mesa-25.0.2-orig/src/nouveau/mme/meson.build mesa-25.0.2/src/nouveau/mme/meson.build
--- mesa-25.0.2-orig/src/nouveau/mme/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/nouveau/mme/meson.build	2025-03-31 13:58:24.030694700 +0800
@@ -25,6 +25,7 @@
     idep_mesautil,
     idep_nvidia_headers,
   ],
+  name_suffix : 'lib',
 )
 
 idep_nouveau_mme = declare_dependency(
diff -urN mesa-25.0.2-orig/src/nouveau/nil/meson.build mesa-25.0.2/src/nouveau/nil/meson.build
--- mesa-25.0.2-orig/src/nouveau/nil/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/nouveau/nil/meson.build	2025-03-31 13:58:24.033438400 +0800
@@ -30,6 +30,7 @@
   include_directories : [inc_include, inc_src],
   dependencies : [idep_nvidia_headers, idep_mesautil],
   gnu_symbol_visibility: 'hidden',
+  name_suffix : 'lib',
 )
 
 _libnil_deps = [
@@ -87,6 +88,7 @@
   _nil_bindings_rs,
   gnu_symbol_visibility: 'hidden',
   rust_abi: 'rust',
+  name_suffix : 'lib',
 )
 
 _libnil = static_library(
@@ -97,6 +99,7 @@
   rust_args: _libnil_rust_args,
   link_with: [_libnil_format_table, _libnil_rs_bindings],
   dependencies: _libnil_deps,
+  name_suffix: 'lib',
 )
 
 _nil_h = custom_target(
diff -urN mesa-25.0.2-orig/src/nouveau/vulkan/meson.build mesa-25.0.2/src/nouveau/vulkan/meson.build
--- mesa-25.0.2-orig/src/nouveau/vulkan/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/nouveau/vulkan/meson.build	2025-03-31 13:58:24.035943600 +0800
@@ -128,6 +128,7 @@
   dependencies : [nvk_deps, idep_nouveau_codegen],
   c_args : [no_override_init_args, nvk_flags],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',
 )
 
 libvulkan_nouveau = shared_library(
diff -urN mesa-25.0.2-orig/src/nouveau/winsys/meson.build mesa-25.0.2/src/nouveau/winsys/meson.build
--- mesa-25.0.2-orig/src/nouveau/winsys/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/nouveau/winsys/meson.build	2025-03-31 13:58:24.037939000 +0800
@@ -22,6 +22,7 @@
     idep_nvidia_headers,
   ],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',
 )
 
 idep_nouveau_ws = declare_dependency(
diff -urN mesa-25.0.2-orig/src/panfrost/compiler/meson.build mesa-25.0.2/src/panfrost/compiler/meson.build
--- mesa-25.0.2-orig/src/panfrost/compiler/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/panfrost/compiler/meson.build	2025-03-31 13:58:24.040955700 +0800
@@ -125,6 +125,7 @@
   c_args : [no_override_init_args],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 libpanfrost_bifrost = static_library(
@@ -136,6 +137,7 @@
   c_args : [no_override_init_args],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 if with_tests
diff -urN mesa-25.0.2-orig/src/panfrost/compiler/valhall/meson.build mesa-25.0.2/src/panfrost/compiler/valhall/meson.build
--- mesa-25.0.2-orig/src/panfrost/compiler/valhall/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/panfrost/compiler/valhall/meson.build	2025-03-31 13:58:24.043946900 +0800
@@ -41,6 +41,7 @@
   c_args : [no_override_init_args],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 if with_tests
diff -urN mesa-25.0.2-orig/src/panfrost/ds/meson.build mesa-25.0.2/src/panfrost/ds/meson.build
--- mesa-25.0.2-orig/src/panfrost/ds/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/panfrost/ds/meson.build	2025-03-31 13:58:24.047912300 +0800
@@ -15,6 +15,7 @@
   sources: pps_panfrost_sources,
   include_directories: pps_panfrost_includes,
   dependencies: [dep_libdrm, dep_perfetto, libpanfrost_dep, dep_panfrost_perf],
+  name_suffix : 'lib',
 )
 
 compile_args_pps_panfrost = ['-DPPS_PANFROST']
diff -urN mesa-25.0.2-orig/src/panfrost/lib/genxml/meson.build mesa-25.0.2/src/panfrost/lib/genxml/meson.build
--- mesa-25.0.2-orig/src/panfrost/lib/genxml/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/panfrost/lib/genxml/meson.build	2025-03-31 13:58:24.050903400 +0800
@@ -29,6 +29,7 @@
     c_args : [no_override_init_args, '-DPAN_ARCH=' + ver],
     gnu_symbol_visibility : 'hidden',
     build_by_default : false,
+    name_suffix : 'lib',
   )
 endforeach
 
@@ -44,4 +45,5 @@
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
   link_with: [libpanfrost_decode_per_arch],
+  name_suffix : 'lib', 
 )
diff -urN mesa-25.0.2-orig/src/panfrost/lib/kmod/meson.build mesa-25.0.2/src/panfrost/lib/kmod/meson.build
--- mesa-25.0.2-orig/src/panfrost/lib/kmod/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/panfrost/lib/kmod/meson.build	2025-03-31 13:58:24.053378300 +0800
@@ -15,6 +15,7 @@
   gnu_symbol_visibility : 'hidden',
   dependencies: [dep_libdrm, idep_mesautil, idep_pan_packers],
   build_by_default : false,
+  name_suffix : 'lib', 
 )
 
 libpankmod_dep = declare_dependency(
diff -urN mesa-25.0.2-orig/src/panfrost/lib/meson.build mesa-25.0.2/src/panfrost/lib/meson.build
--- mesa-25.0.2-orig/src/panfrost/lib/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/panfrost/lib/meson.build	2025-03-31 13:58:24.054600700 +0800
@@ -18,6 +18,7 @@
     c_args : ['-DPAN_ARCH=' + ver],
     gnu_symbol_visibility : 'hidden',
     dependencies : [deps_for_libpanfrost, dep_valgrind],
+    name_suffix : 'lib',
 )
 endforeach
 
@@ -36,6 +37,7 @@
     c_args : ['-DPAN_ARCH=' + ver],
     gnu_symbol_visibility : 'hidden',
     dependencies : [deps_for_libpanfrost, idep_nir],
+    name_suffix : 'lib',
   )
 endforeach
 
@@ -47,6 +49,7 @@
     c_args : ['-DPAN_ARCH=' + ver],
     gnu_symbol_visibility : 'hidden',
     dependencies : [deps_for_libpanfrost, idep_nir],
+    name_suffix : 'lib',
   )
 endforeach
 
@@ -76,6 +79,7 @@
   dependencies: [dep_libdrm, idep_nir, idep_mesautil],
   build_by_default : false,
   link_with: [libpanfrost_pixel_format, libpanfrost_per_arch, libpankmod_lib],
+  name_suffix : 'lib',
 )
 
 libpanfrost_dep = declare_dependency(
diff -urN mesa-25.0.2-orig/src/panfrost/midgard/meson.build mesa-25.0.2/src/panfrost/midgard/meson.build
--- mesa-25.0.2-orig/src/panfrost/midgard/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/panfrost/midgard/meson.build	2025-03-31 13:58:24.057593100 +0800
@@ -45,6 +45,7 @@
   c_args : [no_override_init_args],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib', 
 )
 
 libpanfrost_midgard = static_library(
@@ -59,4 +60,5 @@
   c_args : [no_override_init_args],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/panfrost/perf/meson.build mesa-25.0.2/src/panfrost/perf/meson.build
--- mesa-25.0.2-orig/src/panfrost/perf/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/panfrost/perf/meson.build	2025-03-31 13:58:24.060041300 +0800
@@ -31,7 +31,8 @@
   'panfrost_perf',
   pan_perf_sources,
   include_directories : [inc_include, inc_src, inc_panfrost],
-  dependencies: libpanfrost_dep
+  dependencies: libpanfrost_dep,
+  name_suffix : 'lib',  
 )
 
 dep_panfrost_perf = declare_dependency(
diff -urN mesa-25.0.2-orig/src/panfrost/shared/meson.build mesa-25.0.2/src/panfrost/shared/meson.build
--- mesa-25.0.2-orig/src/panfrost/shared/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/panfrost/shared/meson.build	2025-03-31 13:58:24.062249600 +0800
@@ -18,6 +18,7 @@
   c_args : [no_override_init_args, '-O3'],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib', 
 )
 
 if with_tests
diff -urN mesa-25.0.2-orig/src/panfrost/util/meson.build mesa-25.0.2/src/panfrost/util/meson.build
--- mesa-25.0.2-orig/src/panfrost/util/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/panfrost/util/meson.build	2025-03-31 13:58:24.065240000 +0800
@@ -30,4 +30,5 @@
   c_args : [no_override_init_args],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/panfrost/vulkan/meson.build mesa-25.0.2/src/panfrost/vulkan/meson.build
--- mesa-25.0.2-orig/src/panfrost/vulkan/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/panfrost/vulkan/meson.build	2025-03-31 13:58:24.068047200 +0800
@@ -150,6 +150,7 @@
       dep_valgrind,
     ],
     c_args : [no_override_init_args, panvk_flags, '-DPAN_ARCH=@0@'.format(arch)],
+    name_suffix : 'lib',
   )
 endforeach
 
diff -urN mesa-25.0.2-orig/src/tool/pps/meson.build mesa-25.0.2/src/tool/pps/meson.build
--- mesa-25.0.2-orig/src/tool/pps/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/tool/pps/meson.build	2025-03-31 13:58:24.070562300 +0800
@@ -20,6 +20,7 @@
   sources: pps_sources,
   include_directories: [include_pps, inc_src, pps_includes],
   dependencies: pps_deps,
+  name_suffix : 'lib',
 )
 
 dep_pps = declare_dependency(
diff -urN mesa-25.0.2-orig/src/util/blake3/meson.build mesa-25.0.2/src/util/blake3/meson.build
--- mesa-25.0.2-orig/src/util/blake3/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/util/blake3/meson.build	2025-03-31 13:58:24.072558200 +0800
@@ -40,6 +40,7 @@
   files_blake3,
   c_args : blake3_defs,
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',
 )
 
 idep_blake3 = declare_dependency(
diff -urN mesa-25.0.2-orig/src/util/meson.build mesa-25.0.2/src/util/meson.build
--- mesa-25.0.2-orig/src/util/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/util/meson.build	2025-03-31 13:58:24.077013000 +0800
@@ -271,6 +271,7 @@
   c_args : [c_msvc_compat_args, sse41_args],
   include_directories : [inc_util],
   gnu_symbol_visibility : 'hidden',
+  name_suffix : 'lib',
 )
 
 _libmesa_util = static_library(
@@ -281,7 +282,8 @@
   link_with: [libmesa_util_sse41],
   c_args : [c_msvc_compat_args],
   gnu_symbol_visibility : 'hidden',
-  build_by_default : false
+  build_by_default : false,
+  name_suffix : 'lib',
 )
 
 idep_mesautil = declare_dependency(
@@ -319,6 +321,7 @@
   ],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib', 
 )
 
 idep_xmlconfig = declare_dependency(
@@ -336,6 +339,7 @@
   ['parson.c', 'parson.h'],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 idep_parson = declare_dependency(
diff -urN mesa-25.0.2-orig/src/util/u_gralloc/meson.build mesa-25.0.2/src/util/u_gralloc/meson.build
--- mesa-25.0.2-orig/src/util/u_gralloc/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/util/u_gralloc/meson.build	2025-03-31 13:58:24.080007500 +0800
@@ -44,6 +44,7 @@
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
   dependencies: deps_for_u_gralloc,
+  name_suffix : 'lib',
 )
 
 idep_u_gralloc = declare_dependency(link_with: _libmesa_u_gralloc)
diff -urN mesa-25.0.2-orig/src/virtio/vdrm/meson.build mesa-25.0.2/src/virtio/vdrm/meson.build
--- mesa-25.0.2-orig/src/virtio/vdrm/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/virtio/vdrm/meson.build	2025-03-31 13:58:24.083424000 +0800
@@ -24,4 +24,5 @@
     _libmesa_util,
   ],
   build_by_default: false,
+  name_suffix : 'lib',
 )
diff -urN mesa-25.0.2-orig/src/vulkan/runtime/meson.build mesa-25.0.2/src/vulkan/runtime/meson.build
--- mesa-25.0.2-orig/src/vulkan/runtime/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/vulkan/runtime/meson.build	2025-03-31 13:58:24.086415600 +0800
@@ -206,6 +206,7 @@
   c_args : c_msvc_compat_args,
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 libvulkan_lite_instance = static_library(
@@ -216,6 +217,7 @@
   c_args : ['-DVK_LITE_RUNTIME_INSTANCE=1', c_msvc_compat_args],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib', 
 )
 
 # The sources part is to ensure those generated headers used externally are
@@ -303,6 +305,7 @@
   c_args : c_msvc_compat_args,
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 libvulkan_instance = static_library(
@@ -313,6 +316,7 @@
   c_args : ['-DVK_LITE_RUNTIME_INSTANCE=0', c_msvc_compat_args],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 if get_option('backend').startswith('vs')
diff -urN mesa-25.0.2-orig/src/vulkan/util/meson.build mesa-25.0.2/src/vulkan/util/meson.build
--- mesa-25.0.2-orig/src/vulkan/util/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/vulkan/util/meson.build	2025-03-31 13:58:24.089384100 +0800
@@ -128,6 +128,7 @@
   c_args : [c_msvc_compat_args],
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib',
 )
 
 idep_vulkan_util_headers = declare_dependency(
diff -urN mesa-25.0.2-orig/src/vulkan/wsi/meson.build mesa-25.0.2/src/vulkan/wsi/meson.build
--- mesa-25.0.2-orig/src/vulkan/wsi/meson.build	2025-03-20 21:32:27.000000000 +0800
+++ mesa-25.0.2/src/vulkan/wsi/meson.build	2025-03-31 13:58:24.091970800 +0800
@@ -63,6 +63,7 @@
   link_with: links_vulkan_wsi,
   gnu_symbol_visibility : 'hidden',
   build_by_default : false,
+  name_suffix : 'lib', 
 )
 
 idep_vulkan_wsi_headers = declare_dependency(
