diff -urN glib-2.83.3-orig/gio/inotify/meson.build glib-2.83.3/gio/inotify/meson.build
--- glib-2.83.3-orig/gio/inotify/meson.build	2025-01-23 21:20:40.000000000 +0800
+++ glib-2.83.3/gio/inotify/meson.build	2025-04-07 17:30:46.547649700 +0800
@@ -43,4 +43,5 @@
   ],
   gnu_symbol_visibility : 'hidden',
   pic : true,
+  name_suffix : 'lib',
   c_args : [gio_c_args, gio_c_args_internal])
diff -urN glib-2.83.3-orig/gio/kqueue/meson.build glib-2.83.3/gio/kqueue/meson.build
--- glib-2.83.3-orig/gio/kqueue/meson.build	2025-01-23 21:20:40.000000000 +0800
+++ glib-2.83.3/gio/kqueue/meson.build	2025-04-07 17:30:54.595077100 +0800
@@ -16,4 +16,5 @@
   ],
   gnu_symbol_visibility : 'hidden',
   pic : true,
+  name_suffix : 'lib',
   c_args : [gio_c_args, gio_c_args_internal])
diff -urN glib-2.83.3-orig/gio/win32/meson.build glib-2.83.3/gio/win32/meson.build
--- glib-2.83.3-orig/gio/win32/meson.build	2025-01-23 21:20:40.000000000 +0800
+++ glib-2.83.3/gio/win32/meson.build	2025-04-07 17:31:01.759101600 +0800
@@ -18,4 +18,5 @@
   ],
   gnu_symbol_visibility : 'hidden',
   pic : true,
+  name_suffix : 'lib', 
   c_args : [gio_c_args, gio_c_args_internal])
diff -urN glib-2.83.3-orig/gio/xdgmime/meson.build glib-2.83.3/gio/xdgmime/meson.build
--- glib-2.83.3-orig/gio/xdgmime/meson.build	2025-01-23 21:20:40.000000000 +0800
+++ glib-2.83.3/gio/xdgmime/meson.build	2025-04-07 17:31:08.585092000 +0800
@@ -13,6 +13,7 @@
   sources : xdgmime_sources,
   include_directories : [configinc],
   pic : true,
+  name_suffix : 'lib',
   c_args : [ '-DHAVE_CONFIG_H',
              '-DXDG_PREFIX=_gio_xdg' ],
   gnu_symbol_visibility : 'hidden',
diff -urN glib-2.83.3-orig/girepository/cmph/meson.build glib-2.83.3/girepository/cmph/meson.build
--- glib-2.83.3-orig/girepository/cmph/meson.build	2025-01-23 21:20:40.000000000 +0800
+++ glib-2.83.3/girepository/cmph/meson.build	2025-04-07 17:31:28.526188500 +0800
@@ -50,6 +50,7 @@
 cmph = static_library('cmph',
   sources: cmph_sources,
   c_args: custom_c_args,
+  name_suffix : 'lib',
   dependencies: cmph_deps,
   gnu_symbol_visibility : 'hidden',
 )
diff -urN glib-2.83.3-orig/girepository/meson.build glib-2.83.3/girepository/meson.build
--- glib-2.83.3-orig/girepository/meson.build	2025-01-23 21:20:40.000000000 +0800
+++ glib-2.83.3/girepository/meson.build	2025-04-07 17:31:21.637226800 +0800
@@ -107,6 +107,7 @@
     libgmodule_dep,
     libgobject_dep,
   ],
+  name_suffix : 'lib',
   gnu_symbol_visibility : 'hidden',
 )
 
@@ -133,6 +134,7 @@
     'girwriter.c',
     gi_visibility_h,
   ],
+  name_suffix : 'lib',
   c_args: gir_c_args + custom_c_args,
   gnu_symbol_visibility : 'hidden',
   include_directories : [configinc, girepoinc],
diff -urN glib-2.83.3-orig/glib/gnulib/meson.build glib-2.83.3/glib/gnulib/meson.build
--- glib-2.83.3-orig/glib/gnulib/meson.build	2025-01-23 21:20:40.000000000 +0800
+++ glib-2.83.3/glib/gnulib/meson.build	2025-04-07 17:31:39.262211200 +0800
@@ -382,6 +382,7 @@
   dependencies : [libm],
   include_directories : [configinc, glibinc, include_directories ('.')],
   pic : true,
+  name_suffix : 'lib', 
   c_args : ['-DGCC_LINT=1', '-DLIBDIR="@0@"'.format(glib_libdir), '-DG_LOG_DOMAIN="GLib"',
             glib_c_args_internal, extra_gnulib_args],
   gnu_symbol_visibility : 'hidden',
diff -urN glib-2.83.3-orig/glib/libcharset/meson.build glib-2.83.3/glib/libcharset/meson.build
--- glib-2.83.3-orig/glib/libcharset/meson.build	2025-01-23 21:20:40.000000000 +0800
+++ glib-2.83.3/glib/libcharset/meson.build	2025-04-07 17:31:45.808011800 +0800
@@ -1,6 +1,7 @@
 charset_lib = static_library('charset', 'localcharset.c',
   include_directories : configinc,
   pic : true,
+  name_suffix : 'lib',
   c_args : [ '-DGLIB_CHARSETALIAS_DIR="@0@"'.format(glib_charsetaliasdir) ],
   gnu_symbol_visibility : 'hidden',
 )
