diff -urN avm-8.1.0-orig/build/cmake/pkg_config.cmake avm-8.1.0/build/cmake/pkg_config.cmake
--- avm-8.1.0-orig/build/cmake/pkg_config.cmake	2024-11-15 07:50:55.000000000 +0800
+++ avm-8.1.0/build/cmake/pkg_config.cmake	2024-11-26 13:43:04.166746800 +0800
@@ -61,8 +61,8 @@
 file(APPEND "${pkgconfig_file}" "Conflicts:\n")
 file(APPEND "${pkgconfig_file}" "Libs: -L\${libdir} -l${pkg_name}\n")
 if(CONFIG_MULTITHREAD AND HAVE_PTHREAD_H)
-  file(APPEND "${pkgconfig_file}" "Libs.private: -lm -lpthread\n")
+  file(APPEND "${pkgconfig_file}" "Libs.private: -lpthread\n")
 else()
-  file(APPEND "${pkgconfig_file}" "Libs.private: -lm\n")
+  file(APPEND "${pkgconfig_file}" "Libs.private:\n")
 endif()
 file(APPEND "${pkgconfig_file}" "Cflags: -I\${includedir}\n")
