diff -urN SuiteSparse-7.10.3-orig/GraphBLAS/Config/GraphBLAS.h.in SuiteSparse-7.10.3/GraphBLAS/Config/GraphBLAS.h.in
--- SuiteSparse-7.10.3-orig/GraphBLAS/Config/GraphBLAS.h.in	2025-05-06 06:25:19.000000000 +0800
+++ SuiteSparse-7.10.3/GraphBLAS/Config/GraphBLAS.h.in	2025-06-22 18:13:18.033709700 +0800
@@ -102,7 +102,7 @@
 // GB_GLOBAL: for declaring global variables visible to the user application.
 // These are not used for functions, just global variables like the predefined
 // operators (GrB_PLUS_FP32), types, monoids, semirings, and descriptors.
-#if defined (_MSC_VER) && !(defined (__INTEL_COMPILER) || defined(__INTEL_CLANG_COMPILER))
+#if defined (_MSC_VER)
     #if defined ( GB_DLL_EXPORT )
         // Compiling SuiteSparse:GraphBLAS as a Windows DLL, exporting symbols
         // to user apps.
diff -urN SuiteSparse-7.10.3-orig/GraphBLAS/Source/jit_kernels/include/GB_jit_kernel.h SuiteSparse-7.10.3/GraphBLAS/Source/jit_kernels/include/GB_jit_kernel.h
--- SuiteSparse-7.10.3-orig/GraphBLAS/Source/jit_kernels/include/GB_jit_kernel.h	2025-05-06 06:25:19.000000000 +0800
+++ SuiteSparse-7.10.3/GraphBLAS/Source/jit_kernels/include/GB_jit_kernel.h	2025-06-22 18:14:13.903665200 +0800
@@ -26,7 +26,7 @@
 // for all JIT kernels:  the GB_jit_kernel and GB_jit_query functions must be
 // exported so that GB_jitifyer can find the symbols when loading the kernels.
 #include "include/GB_jit_kernel_proto.h"
-#if defined (_MSC_VER) && !(defined (__INTEL_COMPILER) || defined(__INTEL_CLANG_COMPILER))
+#if defined (_MSC_VER)
     #define GB_JIT_GLOBAL extern __declspec ( dllexport )
 #else
     #define GB_JIT_GLOBAL
