diff -urN plasma-24.8.7-orig/tools/subs.py plasma-24.8.7/tools/subs.py
--- plasma-24.8.7-orig/tools/subs.py	2024-08-08 04:58:08.000000000 +0800
+++ plasma-24.8.7/tools/subs.py	2024-12-30 14:47:15.377209500 +0800
@@ -437,8 +437,8 @@
     #('plasma_s',             'plasma_c'            ),
 
     # ----- Fortran examples
-    ('real\(',               'complex\(',          ),
-    ('\(transpose\(',        'conjg\(transpose\('  ),
+    (r'real\(',             r'complex\(',          ),
+    (r'\(transpose\(',      r'conjg\(transpose\('  ),
 
   ],  # end mixed
 
@@ -463,7 +463,7 @@
     ('symmetric',            'symmetric',            'hermitian',            'hermitian'           ),
     ('symmetric',            'symmetric',            'Hermitian',            'Hermitian'           ),
     ('orthogonal',           'orthogonal',           'unitary',              'unitary'             ),
-    ('\^T',                  '\^T',                  '\^H',                  '\^H'                 ),
+    (r'\^T',                 r'\^T',                 r'\^H',                 r'\^H'                ),
     ('%f',                   '%lf',                  '%f',                   '%lf'                 ),  # for scanf
 
     # ----- CBLAS
@@ -551,8 +551,8 @@
 
     # ----- Fortran examples
     ('wp = sp',              'wp = dp',              'wp = sp',              'wp = dp'             ),
-    ('real\(wp\)',           'real\(wp\)',           'complex\(wp\)',        'complex\(wp\)'       ),
-    ('\(transpose\(',        '\(transpose\(',        'conjg\(transpose\(',   'conjg\(transpose\('  ),
+    (r'real\(wp\)',          r'real\(wp\)',          r'complex\(wp\)',       r'complex\(wp\)'      ),
+    (r'\(transpose\(',       r'\(transpose\(',       r'conjg\(transpose\(',  r'conjg\(transpose\(' ),
 
   ],  # end normal
 } # end subs
