diff -urN gtkmm-4.18.0-orig/untracked/gdk/gdkmm/drag.h gtkmm-4.18.0/untracked/gdk/gdkmm/drag.h
--- gtkmm-4.18.0-orig/untracked/gdk/gdkmm/drag.h	2025-03-24 02:01:48.054945700 +0800
+++ gtkmm-4.18.0/untracked/gdk/gdkmm/drag.h	2025-08-01 12:22:16.224285800 +0800
@@ -6,6 +6,10 @@
 #include <glibmm/ustring.h>
 #include <sigc++/sigc++.h>
 
+#ifdef _MSC_VER
+#undef ERROR
+#endif
+
 /*
  * Copyright (C) 1998-2002 The gtkmm Development Team
  *
diff -urN gtkmm-4.18.0-orig/untracked/gdk/gdkmm/enums.h gtkmm-4.18.0/untracked/gdk/gdkmm/enums.h
--- gtkmm-4.18.0-orig/untracked/gdk/gdkmm/enums.h	2025-03-24 02:01:48.056106300 +0800
+++ gtkmm-4.18.0/untracked/gdk/gdkmm/enums.h	2025-07-31 08:43:16.823040200 +0800
@@ -190,6 +190,9 @@
  *
  * @ingroup gdkmmEnums
  */
+#ifdef _MSC_VER
+#undef IGNORE
+#endif
 enum class AxisUse
 {
   IGNORE,
diff -urN gtkmm-4.18.0-orig/untracked/gdk/gdkmm/graphene_point.h gtkmm-4.18.0/untracked/gdk/gdkmm/graphene_point.h
--- gtkmm-4.18.0-orig/untracked/gdk/gdkmm/graphene_point.h	2025-03-24 02:01:48.057765000 +0800
+++ gtkmm-4.18.0/untracked/gdk/gdkmm/graphene_point.h	2025-08-01 12:23:03.471660000 +0800
@@ -8,6 +8,10 @@
 #include <glibmm/ustring.h>
 #include <sigc++/sigc++.h>
 
+#ifdef _MSC_VER
+#undef near
+#endif
+
 /* Copyright (C) 2023 The gtkmm Development Team
  *
  * This library is free software; you can redistribute it and/or
diff -urN gtkmm-4.18.0-orig/untracked/gtk/gtkmm/enums.h gtkmm-4.18.0/untracked/gtk/gtkmm/enums.h
--- gtkmm-4.18.0-orig/untracked/gtk/gtkmm/enums.h	2025-03-24 02:01:48.116617400 +0800
+++ gtkmm-4.18.0/untracked/gtk/gtkmm/enums.h	2025-08-01 12:32:28.976564800 +0800
@@ -8,6 +8,10 @@
 #include <glibmm/ustring.h>
 #include <sigc++/sigc++.h>
 
+#ifdef _MSC_VER
+#undef ERROR
+#endif
+
 /*
  * Copyright (C) 2002 The gtkmm Development Team
  *
