diff -urN liboggz-1.0.2-orig/src/tools/oggz-comment.c liboggz-1.0.2/src/tools/oggz-comment.c
--- liboggz-1.0.2-orig/src/tools/oggz-comment.c	2010-04-29 09:44:08.000000000 +0800
+++ liboggz-1.0.2/src/tools/oggz-comment.c	2025-02-20 10:54:28.290926000 +0800
@@ -51,6 +51,10 @@
 
 #define S_SERIALNO 0x7
 
+#ifdef WIN32                                                                   
+#define strcasecmp _stricmp
+#endif
+
 typedef struct {
   int do_delete;
   int do_all;
