diff -urN cairo-1.18.2-orig/boilerplate/meson.build cairo-1.18.2/boilerplate/meson.build
--- cairo-1.18.2-orig/boilerplate/meson.build	2024-09-01 21:09:23.000000000 +0800
+++ cairo-1.18.2/boilerplate/meson.build	2025-04-08 17:18:49.652371700 +0800
@@ -31,6 +31,7 @@
   c_args: ['-DCAIRO_COMPILATION'],
   install: false,
   build_by_default: false,
+  name_suffix: 'lib',
 )
 
 cairoboilerplate_dep = declare_dependency(
diff -urN cairo-1.18.2-orig/perf/meson.build cairo-1.18.2/perf/meson.build
--- cairo-1.18.2-orig/perf/meson.build	2024-09-01 21:09:23.000000000 +0800
+++ cairo-1.18.2/perf/meson.build	2025-04-08 17:19:05.576720200 +0800
@@ -15,6 +15,7 @@
   ],
   include_directories: [incbase, incsrc],
   dependencies: [pixman_dep, libcairomissing_dep, cairoboilerplate_dep],
+  name_suffix: 'lib',
 )
 libcairoperf_dep = declare_dependency(
   link_with: libcairoperf,
diff -urN cairo-1.18.2-orig/perf/micro/meson.build cairo-1.18.2/perf/micro/meson.build
--- cairo-1.18.2-orig/perf/micro/meson.build	2024-09-01 21:09:23.000000000 +0800
+++ cairo-1.18.2/perf/micro/meson.build	2025-04-08 17:19:12.168258300 +0800
@@ -54,6 +54,7 @@
   perf_micro_sources + perf_micro_headers,
   include_directories: [incbase, incsrc, incmicro],
   dependencies: [pixman_dep, cairoboilerplate_dep],
+  name_suffix: 'lib',
 )
 libcairoperfmicro_dep = declare_dependency(
   link_with: libcairoperfmicro,
diff -urN cairo-1.18.2-orig/test/pdiff/meson.build cairo-1.18.2/test/pdiff/meson.build
--- cairo-1.18.2-orig/test/pdiff/meson.build	2024-09-01 21:09:23.000000000 +0800
+++ cairo-1.18.2/test/pdiff/meson.build	2025-04-08 17:19:18.763227500 +0800
@@ -11,6 +11,7 @@
 libpdiff = static_library('pdiff', libpdiff_sources,
   include_directories: [incbase],
   dependencies: deps + [libcairo_dep],
+  name_suffix: 'lib',
 )
 
 libpdiff_dep = declare_dependency(
diff -urN cairo-1.18.2-orig/util/cairo-missing/meson.build cairo-1.18.2/util/cairo-missing/meson.build
--- cairo-1.18.2-orig/util/cairo-missing/meson.build	2024-09-01 21:09:23.000000000 +0800
+++ cairo-1.18.2/util/cairo-missing/meson.build	2025-04-08 17:19:24.971254900 +0800
@@ -7,6 +7,7 @@
   include_directories: [incbase, incsrc],
   install: false,
   build_by_default: false,
+  name_suffix: 'lib',
 )
 
 libcairomissing_dep = declare_dependency(
