diff -urN cairo-1.18.2-orig/src/cairo-atomic-private.h cairo-1.18.2/src/cairo-atomic-private.h
--- cairo-1.18.2-orig/src/cairo-atomic-private.h	2024-09-01 21:09:23.000000000 +0800
+++ cairo-1.18.2/src/cairo-atomic-private.h	2025-04-08 17:25:22.287199100 +0800
@@ -291,7 +291,11 @@
 
 typedef intptr_t cairo_atomic_intptr_t;
 
+#ifdef _MSC_VER
+# define _cairo_atomic_ptr_get(x) _cairo_atomic_intptr_to_voidptr (AO_load_full ((volatile const size_t *)x))
+#else
 # define _cairo_atomic_ptr_get(x) _cairo_atomic_intptr_to_voidptr (AO_load_full (x))
+#endif
 # define _cairo_atomic_ptr_cmpxchg(x, oldv, newv) \
     _cairo_atomic_int_cmpxchg ((cairo_atomic_intptr_t*)(x), (cairo_atomic_intptr_t)oldv, (cairo_atomic_intptr_t)newv)
 
