diff -urN gstreamer-1.26.0-orig/gst/printf/meson.build gstreamer-1.26.0/gst/printf/meson.build
--- gstreamer-1.26.0-orig/gst/printf/meson.build	2025-03-12 04:14:44.000000000 +0800
+++ gstreamer-1.26.0/gst/printf/meson.build	2025-04-07 17:24:24.049605600 +0800
@@ -81,4 +81,5 @@
     c_args : printf_args,
     install : false,
     pic: true,
+    name_suffix: 'lib',
     dependencies : [glib_dep])
diff -urN gstreamer-1.26.0-orig/libs/gst/check/libcheck/meson.build gstreamer-1.26.0/libs/gst/check/libcheck/meson.build
--- gstreamer-1.26.0-orig/libs/gst/check/libcheck/meson.build	2025-03-12 04:14:44.000000000 +0800
+++ gstreamer-1.26.0/libs/gst/check/libcheck/meson.build	2025-04-07 17:24:33.105845400 +0800
@@ -86,4 +86,5 @@
 	  # Don't want libcompat to think we don't have these and substitute
 	  # replacements since we don't check for or define these. See libcompat.h
 	  ['-DHAVE_VSNPRINTF', '-DHAVE_SNPRINTF', '-DHAVE_MALLOC', '-DHAVE_REALLOC'],
+  name_suffix: 'lib',
   pic: true)
diff -urN gstreamer-1.26.0-orig/libs/gst/helpers/ptp/meson.build gstreamer-1.26.0/libs/gst/helpers/ptp/meson.build
--- gstreamer-1.26.0-orig/libs/gst/helpers/ptp/meson.build	2025-03-12 04:14:44.000000000 +0800
+++ gstreamer-1.26.0/libs/gst/helpers/ptp/meson.build	2025-04-07 17:24:44.493695100 +0800
@@ -98,6 +98,7 @@
 
 conf = static_library('gst_ptp_helper_conf', conf_lib_rs,
   override_options : ['rust_std=2018'],
+  name_suffix: 'lib',
   rust_crate_type : 'rlib')
 
 exe = executable('gst-ptp-helper', 'main.rs',
