diff -urN pangomm-2.56.1-orig/untracked/pango/pangomm/attributes.h pangomm-2.56.1/untracked/pango/pangomm/attributes.h
--- pangomm-2.56.1-orig/untracked/pango/pangomm/attributes.h	2025-01-16 02:44:11.631735300 +0800
+++ pangomm-2.56.1/untracked/pango/pangomm/attributes.h	2025-04-12 22:26:06.319345800 +0800
@@ -273,6 +273,9 @@
  *
  * @ingroup pangommEnums
  */
+#ifdef _MSC_VER
+#undef ERROR
+#endif
 enum class Underline
 {
   NONE,
