diff -urN vmaf-3.0.0-orig/libvmaf/src/feature/cuda/integer_adm_cuda.c vmaf-3.0.0/libvmaf/src/feature/cuda/integer_adm_cuda.c
--- vmaf-3.0.0-orig/libvmaf/src/feature/cuda/integer_adm_cuda.c	2023-12-08 03:27:32.000000000 +0800
+++ vmaf-3.0.0/libvmaf/src/feature/cuda/integer_adm_cuda.c	2025-02-12 11:51:20.988338000 +0800
@@ -30,7 +30,11 @@
 #include "cpu.h"
 #include "cuda/integer_adm_cuda.h"
 #include "picture_cuda.h"
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#else
+#include <windows.h>
+#endif
 
 #include <assert.h>
 
diff -urN vmaf-3.0.0-orig/libvmaf/src/feature/integer_adm.c vmaf-3.0.0/libvmaf/src/feature/integer_adm.c
--- vmaf-3.0.0-orig/libvmaf/src/feature/integer_adm.c	2023-12-08 03:27:32.000000000 +0800
+++ vmaf-3.0.0/libvmaf/src/feature/integer_adm.c	2025-02-12 11:51:20.994107700 +0800
@@ -22,6 +22,7 @@
 #include "feature_extractor.h"
 #include "feature_name.h"
 #include "integer_adm.h"
+#include "integer_vif.h"
 #include "log.h"
 
 #if ARCH_X86
diff -urN vmaf-3.0.0-orig/libvmaf/src/feature/integer_vif.h vmaf-3.0.0/libvmaf/src/feature/integer_vif.h
--- vmaf-3.0.0-orig/libvmaf/src/feature/integer_vif.h	2023-12-08 03:27:32.000000000 +0800
+++ vmaf-3.0.0/libvmaf/src/feature/integer_vif.h	2025-02-12 11:51:21.006218700 +0800
@@ -130,6 +130,7 @@
 #ifdef _MSC_VER
 #include <intrin.h>
 
+# ifndef __clang__
 static inline int __builtin_clz(unsigned x) {
     return (int)__lzcnt(x);
 }
@@ -137,6 +138,7 @@
 static inline int __builtin_clzll(unsigned long long x) {
     return (int)__lzcnt64(x);
 }
+# endif
 
 #endif
 
diff -urN vmaf-3.0.0-orig/libvmaf/src/feature/mkdirp.c vmaf-3.0.0/libvmaf/src/feature/mkdirp.c
--- vmaf-3.0.0-orig/libvmaf/src/feature/mkdirp.c	2023-12-08 03:27:32.000000000 +0800
+++ vmaf-3.0.0/libvmaf/src/feature/mkdirp.c	2025-02-12 11:51:21.011248800 +0800
@@ -6,7 +6,11 @@
 // MIT licensed
 //
 
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#else
+#include <windows.h>
+#endif
 #include <errno.h>
 #include <stdlib.h>
 #include <string.h>
diff -urN vmaf-3.0.0-orig/libvmaf/src/feature/mkdirp.h vmaf-3.0.0/libvmaf/src/feature/mkdirp.h
--- vmaf-3.0.0-orig/libvmaf/src/feature/mkdirp.h	2023-12-08 03:27:32.000000000 +0800
+++ vmaf-3.0.0/libvmaf/src/feature/mkdirp.h	2025-02-12 11:51:21.014240700 +0800
@@ -12,6 +12,10 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 
+#ifdef _MSC_VER
+typedef unsigned short mode_t;
+#endif
+
 /*
  * Recursively `mkdir(path, mode)`
  */
diff -urN vmaf-3.0.0-orig/libvmaf/src/log.c vmaf-3.0.0/libvmaf/src/log.c
--- vmaf-3.0.0-orig/libvmaf/src/log.c	2023-12-08 03:27:32.000000000 +0800
+++ vmaf-3.0.0/libvmaf/src/log.c	2025-02-12 11:51:21.016235600 +0800
@@ -19,7 +19,11 @@
 #include "libvmaf/libvmaf.h"
 
 #include <stdarg.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#else
+#include <windows.h>
+#endif
 
 static enum VmafLogLevel vmaf_log_level = VMAF_LOG_LEVEL_INFO;
 static int istty = 0;
diff -urN vmaf-3.0.0-orig/libvmaf/test/test_ring_buffer.c vmaf-3.0.0/libvmaf/test/test_ring_buffer.c
--- vmaf-3.0.0-orig/libvmaf/test/test_ring_buffer.c	2023-12-08 03:27:32.000000000 +0800
+++ vmaf-3.0.0/libvmaf/test/test_ring_buffer.c	2025-02-12 11:51:21.017256500 +0800
@@ -20,7 +20,11 @@
 #include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#else
+#include <windows.h>
+#endif
 
 #include "test.h"
 
diff -urN vmaf-3.0.0-orig/libvmaf/tools/vmaf.c vmaf-3.0.0/libvmaf/tools/vmaf.c
--- vmaf-3.0.0-orig/libvmaf/tools/vmaf.c	2023-12-08 03:27:32.000000000 +0800
+++ vmaf-3.0.0/libvmaf/tools/vmaf.c	2025-02-12 11:51:21.017256500 +0800
@@ -1,7 +1,11 @@
 #include <stdlib.h>
 #include <string.h>
 #include <time.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#else
+#include <windows.h>
+#endif
 
 #include "cli_parse.h"
 #include "spinner.h"
