diff -urN vmaf-3.0.0-orig/libvmaf/src/meson.build vmaf-3.0.0/libvmaf/src/meson.build
--- vmaf-3.0.0-orig/libvmaf/src/meson.build	2023-12-08 03:27:32.000000000 +0800
+++ vmaf-3.0.0/libvmaf/src/meson.build	2025-02-16 16:07:16.059118200 +0800
@@ -208,6 +208,7 @@
     'libvmaf_cpu',
     libvmaf_cpu_sources,
     include_directories : [libvmaf_include],
+    name_suffix : 'lib',
 )
 
 platform_specific_cpu_objects = []
@@ -223,7 +224,8 @@
           'arm64_v8',
           arm64_sources,
           include_directories : vmaf_base_include,
-          c_args : vmaf_cflags_common + ['-DARCH_AARCH64']
+          c_args : vmaf_cflags_common + ['-DARCH_AARCH64'],
+          name_suffix : 'lib',
         )
 
         platform_specific_cpu_objects += arm64_static_lib.extract_all_objects()
@@ -243,6 +245,7 @@
           x86_avx2_sources,
           include_directories : vmaf_base_include,
           c_args : ['-mavx', '-mavx2'] + vmaf_cflags_common,
+          name_suffix : 'lib',
       )
 
       platform_specific_cpu_objects += x86_avx2_static_lib.extract_all_objects(recursive: true)
@@ -260,6 +263,7 @@
             c_args : ['-mavx512f', '-mavx512dq', '-mavx512bw', '-mavx512cd', '-mavx512dq',
                       '-mavx512vbmi', '-mavx512vl'] +
                      vmaf_cflags_common,
+            name_suffix : 'lib',
         )
 
         platform_specific_cpu_objects += x86_avx512_static_lib.extract_all_objects(recursive: true)
@@ -359,6 +363,7 @@
             include_directories('../src/cuda/'),
             cuda_inc,
         ],
+        name_suffix : 'lib',
         c_args : vmaf_cflags_common,
         cuda_args: cuda_flags # + ['-gencode', 'arch=compute_86,code=sm_86' ] #, '--use_fast_math']
     )
@@ -442,7 +447,8 @@
     libvmaf_feature_sources,
     include_directories : [libvmaf_include, vmaf_include, cuda_dir],
     dependencies: [stdatomic_dependency, cuda_dependency],
-    objects: common_cuda_objects
+    objects: common_cuda_objects,
+    name_suffix : 'lib',
 )
 
 libvmaf_sources = [
@@ -496,6 +502,7 @@
     version : vmaf_soname_version,
     soversion : vmaf_soversion,
     install: true,
+    name_suffix: 'lib',
 )
 
 pkg_mod = import('pkgconfig')
