diff -urN gtk-4.18.3-orig/gdk/meson.build gtk-4.18.3/gdk/meson.build
--- gtk-4.18.3-orig/gdk/meson.build	2025-03-29 20:45:29.000000000 +0800
+++ gtk-4.18.3/gdk/meson.build	2025-04-12 15:11:25.029429900 +0800
@@ -296,6 +296,7 @@
 libgdk = static_library('gdk',
   sources: [gdk_sources, gdk_backends_gen_headers, gdk_gen_headers],
   dependencies: gdk_deps + [libgtk_css_dep],
+  name_suffix: 'lib',
   link_with: [libgtk_css],
   include_directories: [confinc, gdkx11_inc, wlinc],
   c_args: libgdk_c_args + common_cflags,
diff -urN gtk-4.18.3-orig/gdk/win32/meson.build gtk-4.18.3/gdk/win32/meson.build
--- gtk-4.18.3-orig/gdk/win32/meson.build	2025-03-29 20:45:29.000000000 +0800
+++ gtk-4.18.3/gdk/win32/meson.build	2025-04-12 15:11:25.046531500 +0800
@@ -70,4 +70,5 @@
     '-DINSIDE_GDK_WIN32',
   ],
   dependencies: [ gdk_deps, gdk_win32_deps ],
+  name_suffix: 'lib',
 )
diff -urN gtk-4.18.3-orig/gsk/meson.build gtk-4.18.3/gsk/meson.build
--- gtk-4.18.3-orig/gsk/meson.build	2025-03-29 20:45:29.000000000 +0800
+++ gtk-4.18.3/gsk/meson.build	2025-04-12 15:39:47.873873100 +0800
@@ -181,6 +181,7 @@
 libgsk_f16c = static_library('gsk_f16c',
   sources: gsk_f16c_sources,
   dependencies: gsk_deps,
+  name_suffix: 'lib',
   include_directories: [ confinc, ],
   c_args: [
     '-DGTK_COMPILATION',
@@ -199,6 +200,7 @@
     gsk_private_gpu_shader_headers,
   ],
   dependencies: gsk_deps,
+  name_suffix: 'lib',
   include_directories: [ confinc, ],
   c_args: [
     '-DGTK_COMPILATION',
diff -urN gtk-4.18.3-orig/gtk/css/meson.build gtk-4.18.3/gtk/css/meson.build
--- gtk-4.18.3-orig/gtk/css/meson.build	2025-03-29 20:45:29.000000000 +0800
+++ gtk-4.18.3/gtk/css/meson.build	2025-04-12 15:11:25.063325400 +0800
@@ -46,6 +46,7 @@
     gdk_visibility_h,
   ],
   dependencies: gtk_css_deps,
+  name_suffix: 'lib',
   include_directories: [ confinc, ],
   c_args: [
     '-DGTK_COMPILATION',
diff -urN gtk-4.18.3-orig/gtk/meson.build gtk-4.18.3/gtk/meson.build
--- gtk-4.18.3-orig/gtk/meson.build	2025-03-29 20:45:29.000000000 +0800
+++ gtk-4.18.3/gtk/meson.build	2025-04-12 15:11:25.079540000 +0800
@@ -1122,6 +1122,7 @@
   c_args: gtk_cargs + common_cflags,
   include_directories: [confinc, gdkinc, gskinc, gtkinc],
   dependencies: gtk_deps + [libgtk_css_dep, libgdk_dep, libgsk_dep],
+  name_suffix: 'lib',
   link_with: [libgtk_css, libgdk, libgsk ],
 )
 
