diff -urN Support-orig/CMakeLists.txt Support/CMakeLists.txt
--- Support-orig/CMakeLists.txt	2025-03-19 07:15:15.000000000 +0800
+++ Support/CMakeLists.txt	2025-04-04 11:17:38.435425300 +0800
@@ -332,10 +332,18 @@
   # CMAKE_BUILD_TYPE is only meaningful to single-configuration generators.
   if(CMAKE_BUILD_TYPE)
     string(TOUPPER ${CMAKE_BUILD_TYPE} build_type)
-    get_property(zstd_library TARGET ${zstd_target} PROPERTY LOCATION_${build_type})
+    if(MSVC)
+      get_property(zstd_library TARGET ${zstd_target} PROPERTY IMPORTED_IMPLIB_${build_type})
+    else()
+      get_property(zstd_library TARGET ${zstd_target} PROPERTY LOCATION_${build_type})
+    endif()
   endif()
   if(NOT zstd_library)
-    get_property(zstd_library TARGET ${zstd_target} PROPERTY LOCATION)
+    if(MSVC)
+      get_property(zstd_library TARGET ${zstd_target} PROPERTY IMPORTED_IMPLIB)
+    else()
+      get_property(zstd_library TARGET ${zstd_target} PROPERTY LOCATION)
+    endif()
   endif()
   if (zstd_target STREQUAL zstd::libzstd_shared)
     get_library_name(${zstd_library} zstd_library)
