diff -urN openh264-2.5.0-orig/codec/common/meson.build openh264-2.5.0/codec/common/meson.build
--- openh264-2.5.0-orig/codec/common/meson.build	2024-11-06 14:05:33.000000000 +0800
+++ openh264-2.5.0/codec/common/meson.build	2025-04-08 20:13:39.361492400 +0800
@@ -77,4 +77,5 @@
 
 libcommon = static_library('common', cpp_sources, objs_asm,
   include_directories: [inc, casm_inc],
+  name_suffix: 'lib',
   dependencies: deps)
diff -urN openh264-2.5.0-orig/codec/console/common/meson.build openh264-2.5.0/codec/console/common/meson.build
--- openh264-2.5.0-orig/codec/console/common/meson.build	2024-11-06 14:05:33.000000000 +0800
+++ openh264-2.5.0/codec/console/common/meson.build	2025-04-08 20:13:46.482248800 +0800
@@ -9,4 +9,5 @@
 
 libconsole_common = static_library('console_common', cpp_sources,
   include_directories: conscomminc,
+  name_suffix: 'lib',
   dependencies: deps)
diff -urN openh264-2.5.0-orig/codec/decoder/meson.build openh264-2.5.0/codec/decoder/meson.build
--- openh264-2.5.0-orig/codec/decoder/meson.build	2024-11-06 14:05:33.000000000 +0800
+++ openh264-2.5.0/codec/decoder/meson.build	2025-04-08 20:13:53.092288000 +0800
@@ -62,4 +62,5 @@
 
 libdecoder = static_library('decoder', cpp_sources, objs_asm,
   include_directories: [inc, decoder_inc, casm_inc],
+  name_suffix: 'lib',
   dependencies: deps)
diff -urN openh264-2.5.0-orig/codec/encoder/meson.build openh264-2.5.0/codec/encoder/meson.build
--- openh264-2.5.0-orig/codec/encoder/meson.build	2024-11-06 14:05:33.000000000 +0800
+++ openh264-2.5.0/codec/encoder/meson.build	2025-04-08 20:14:06.741973800 +0800
@@ -91,4 +91,5 @@
 
 libencoder = static_library('encoder', cpp_sources, objs_asm,
   include_directories: [inc, processing_inc, encoder_inc, casm_inc],
+  name_suffix: 'lib',
   dependencies: deps)
diff -urN openh264-2.5.0-orig/codec/processing/meson.build openh264-2.5.0/codec/processing/meson.build
--- openh264-2.5.0-orig/codec/processing/meson.build	2024-11-06 14:05:33.000000000 +0800
+++ openh264-2.5.0/codec/processing/meson.build	2025-04-08 20:14:11.726417300 +0800
@@ -64,4 +64,5 @@
 
 libprocessing = static_library('processing', cpp_sources, objs_asm,
   include_directories: [inc, processing_inc, casm_inc],
+  name_suffix: 'lib',
   dependencies: deps)
