diff -urN ntldd-0.2-orig/libntldd.c ntldd-0.2/libntldd.c
--- ntldd-0.2-orig/libntldd.c	2015-04-25 15:37:26.000000000 +0800
+++ ntldd-0.2/libntldd.c	2025-10-16 15:33:35.765922200 +0800
@@ -53,7 +53,7 @@
       if (section != NULL)
         *section = i;
       if (soffs[i].off)
-        return soffs[i].off + in_ptr;
+        return (char *)soffs[i].off + in_ptr;  // Cast to char* for byte arithmetic
     }
   return NULL;
 }
