diff -urN glibmm-2.83.1-orig/untracked/gio/giomm/converter.h glibmm-2.83.1/untracked/gio/giomm/converter.h
--- glibmm-2.83.1-orig/untracked/gio/giomm/converter.h	2025-02-11 23:51:19.773518000 +0800
+++ glibmm-2.83.1/untracked/gio/giomm/converter.h	2025-04-12 22:36:12.993445300 +0800
@@ -150,6 +150,9 @@
    *
    * @ingroup giommEnums
    */
+#ifdef _MSC_VER
+#undef ERROR
+#endif
   enum class Result
   {
     ERROR,
diff -urN glibmm-2.83.1-orig/untracked/gio/giomm/dbusmessage.h glibmm-2.83.1/untracked/gio/giomm/dbusmessage.h
--- glibmm-2.83.1-orig/untracked/gio/giomm/dbusmessage.h	2025-02-11 23:51:19.774518300 +0800
+++ glibmm-2.83.1/untracked/gio/giomm/dbusmessage.h	2025-04-12 22:36:54.289543600 +0800
@@ -80,6 +80,9 @@
  *
  * @ingroup giommEnums
  */
+#ifdef _MSC_VER
+#undef ERROR
+#endif
 enum class MessageType
 {
   INVALID,
@@ -938,7 +941,7 @@
   guchar* to_blob(gsize& out_size, CapabilityFlags capabilities =  CapabilityFlags::NONE);
 
   
-  /** If @a message is not of type Gio::DBus::MessageType::ERROR does
+  /** If @a message is not of type Gio::DBus::MessageType::_ERROR does
    * nothing and returns <tt>false</tt>.
    * 
    * Otherwise this method encodes the error in @a message as a Error
diff -urN glibmm-2.83.1-orig/untracked/glib/glibmm/iochannel.h glibmm-2.83.1/untracked/glib/glibmm/iochannel.h
--- glibmm-2.83.1-orig/untracked/glib/glibmm/iochannel.h	2025-02-11 23:51:19.731517800 +0800
+++ glibmm-2.83.1/untracked/glib/glibmm/iochannel.h	2025-04-12 22:37:21.120526900 +0800
@@ -80,6 +80,9 @@
  *
  * @ingroup glibmmEnums
  */
+#ifdef _MSC_VER
+#undef ERROR
+#endif
 enum class IOStatus
 {
   ERROR,
