diff -urN libev-4.33-orig/event.h libev-4.33/event.h
--- libev-4.33-orig/event.h	2018-10-25 11:10:51.000000000 +0800
+++ libev-4.33/event.h	2025-01-07 16:20:34.074781900 +0800
@@ -64,6 +64,8 @@
 #if !defined (WIN32) || defined (__MINGW32__)
 # include <time.h> /* mingw seems to need this, for whatever reason */
 # include <sys/time.h>
+#elif defined (_MSC_VER)
+# include <windows.h>
 #endif
 
 struct event_base;
